# Mapping global shifts in Saccharomyces cerevisiae gene expression across asynchronous time trajectories {#chp-map}

```{=latex}
\begin{epigraphs}
  \qitem{They were maps that lived, maps that one could study, frown over, and add to; maps, in short, that really meant something.}%
        {Gerald Durrell}
\end{epigraphs}
```

```{r libsdm, echo = F, message = F, warning = F}
library(kableExtra)
```


## Abstract

Microbial metabolism is central to ethanol production and the generation or modification of wine flavor compounds during winemaking.
Given that grapes grown in a particular geographic region often produce wines with consistent characteristics, it follows that differences in grape must composition and microbial activities are key to these outcomes. 
However, our current understanding of the factors responsible for these varied fermentation outcomes are not well described. 
To understand how vineyard site location impacts wine fermentation outcomes, we used differences in *Saccharomyces cerevisiae* gene expression as a biosensor for differences among Pinot noir fermentations.
We profiled time series gene expression of primary fermentations using grapes from 15 vineyards and two vintages.
Fermentations proceeded at different rates, making it difficult to analyze these data with conventional differential expression tools. 
This led us to develop a new approach for studying gene expression during global transcriptome shifts by combining diffusion mapping with continuous differential expression analysis. 
Using this method, we identified deviations in the gene expression program in a vineyard-site specific manner, including in genes linked to the aroma profile of wines. 
These gene expression changes correlated with the activity of the non-*Saccharomyces* yeast *Hanseniaspora uvarum*, as well as with initial nitrogen concentration in grape must. 

## Introduction

During a wine fermentation, *Saccharomyces cerevisiae* metabolizes sugars and other nutrients to obtain energy and increase biomass, which results in ethanol and aroma compound production. 
In this way, the fundamental aspects of a wine fermentation are conserved, which is reflected in a core gene expression program (CGEP) that is consistent across diverse fermentation conditions [@rossignol2003genome; @marks2008dynamics; @rossouw2012effect] (see Chapter [1](#chp-brix)).
Yet, *S. cerevisiae* metabolism is not fixed and dynamically responds to differences in the fermentation environment (e.g., nutrient levels, temperature, and microbial communities [@rossignol2003genome; @marks2008dynamics; @molina2007influence; @curiel2017different)) to adapt to the specific conditions of each must.
For example, differences in grape must nitrogen concentrations leads to changes in metabolism that are accompanied by altered aroma compound production [@carrau2008production]. 
This highlights the fact that metabolic adaptation to varied fermentation environments leads to differences in wine fermentation outcomes, including sensory differences. 
This relationship is mirrored by many findings showing that genetic changes causing altered expression of select genes or pathways in *S. cerevisiae* leads to quantifiable differences in wine outcomes [@rossouw2008linking]. 
These facts support the generally accepted idea that interactions between *S. cerevisiae* and the unique chemical and biological matrix of each grape must are central to defining primary fermentation outcomes.
We reason that these changes in outcome are the result of: (1) the expression of unique genes outside those in the CGEP required for fermentation and/or (2) variation in expression of CGEP genes that changes when and how active the various core pathways are during fermentation. 

The chemical and biological diversity of grape musts is in large part a reflection of biotic and abiotic pressures encountered by a grapevine during a growing season. 
For example, wines produced under similar vinification conditions from genetically identical grapes grown in different locations have diverse sensory outcomes [@cantu2020investigating], many of which are reproducible across multiple vintages [@roullier2014grape]. 
This phenomenon is referred to as terroir – the environmental and anthropomorphic factors that contribute to wine outcomes. 
After observing diverse sensory outcomes in wines where the only consistent variable between fermentations was vineyard location [@cantu2020investigating], we sought to uncover quantifiable contributions of vineyard site to wine outcomes by using *S. cerevisiae* gene expression as a biosensor to detect differences between fermentations. 
This is motivated by the fact that high throughput gene expression surveys (microarray, RNA sequencing) have been used for many wine studies and revealed the causes of stuck and sluggish fermentations [@duc2017set], the trigger for entry into stationary phase [@rossignol2003genome; @marks2008dynamics], and the impact of inter-species interactions on *S. cerevisiae* metabolism in wine [@tronchoni2017early; @curiel2017different; @alonso2019dominance].
In addition, as a model organism used for basic science and biotechnology, the *S. cerevisiae* genome and transcriptome is well understood and studies have focused on gene expression in diverse environments, including wine [@cherry1998sgd; @gasch2000genomic; @causton2001remodeling; @rossignol2003genome; @marks2008dynamics; @rossouw2012effect]
We expect this makes *S. cerevisiae* a powerful tool for identifying key biotic and abiotic factors underlying wine fermentation outcomes. 

Towards this end, we performed time series RNA sequencing on Pinot noir fermentations with the aim of identifying gene expression differences due to vineyard site. 
However, using standard analysis methods [@conesa2006masigpro; @robinson2010edger; @love2014moderated; @limma2015; @jung2017timesvector; @abu2018clust], we only identified the dominant and consistent global shift in gene expression, the CGEP, across fermentations and not gene expression patterns that indicated altered *S. cerevisiae* metabolism that would differentiate vineyard site (see Chapter [1](#chp-brix)). 
A major issue was that sampled fermentations proceeded at different rates, leading to asynchronous biological progression among sequenced samples with respect to the time of sampling. 
This was problematic because samples need to be at the same stage of fermentation to interpret the biological significance of differentially expressed genes [@rossouw2012effect; @bar2012studying]. 
This is a common problem in time series experiments with multiple groups and in some experimental systems there are strategies to combat this issue [@bar2012studying].
For example, in experiments that study the cell cycle, inhibitors arrest the cell cycle at the same stage across groups thereby enabling comparisons [@spellman1998comprehensive]. 
This experimental approach is not applicable to wine fermentations where nutrient availability is more important than stage of the cell cycle [@bisson2019gene].

To address a similar issue as we face here, methods have recently been developed for the analysis of single cell RNA sequencing data from differentiating cells.
In these experiments, as cells differentiate, absolute time may not reflect the extent of differentiation in each individual cell [@destiny2015].
Consequently, pseudotime analysis has been used to reorder asynchronous cells from absolute time to the stage in differentiation relative to other cells undergoing the same process [@destiny2015]. 
In particular, diffusion maps have been used to reorder asynchronous cell populations because this analysis approach preserves relationships between samples [@destiny2015]. 
Diffusion mapping is a manifold learning technique that uses information from the k most similar samples to construct non-linear composites of the major sources of variation among samples [@coifman2005geometric; @coifman2006diffusion]. 
Diffusion mapping has strengths over other dimensionality reduction algorithms, like principal component analysis (PCA) and t-distributed stochastic neighbor embedding (tSNE) that are applied to sequencing data to identify sources of variation [@way2020compressing; @moon2019visualizing]. 
The major advantages are that diffusion mapping is non-linear and robust to the “horseshoe effect” (unlike PCA) [@diaconis2008horseshoes], preserves distances between samples (unlike tSNE), and is insensitive to sampling density. 
As a dimensionality reduction algorithm, diffusion maps extract latent variables that are inferred from relationships in the data, which can be used to represent composite sources of variation between samples.
Here, we use diffusion mapping to analyze time-series RNA-sequencing data from *S. cerevisiae* during wine fermentation. 
We use the resulting diffusion maps to synchronize gene expression across treatment groups and to extract latent variables, termed diffusion components (DCs), which represent the dominant sources of structure in the data sets. 
Diffusion maps per se provide no suggestion of the underlying genes that lead to separation of samples along diffusion components; therefore, we apply continuous differential expression analysis using each diffusion component to determine how genes vary among samples. 
We find that diffusion mapping extracted the CGEP across Pinot noir fermentations in 2017 and 2019, in addition to distinguishing subtler differences between fermentations that appear to reflect differences in the grape musts and offer insight into the variable fermentation and sensory outcomes.

Notably, this method is not exclusively useful for the analysis of RNA sequencing from wine fermentations and can be applied in other systems where asynchronous progression of groups with respect to sampling time masks true differences between groups. 
As a proof of principle, we additionally apply this method to time series RNA sequencing data collected from *S. cerevisiae* during hypoxia. 
We show that, as in wine, diffusion mapping captures the dominant global shift in gene expression that occurs when yeast transitions from aerobic to anaerobic metabolism. 
These findings suggest that our method enables analysis of diverse asynchronous time series gene expression data, revealing both global shifts and subtle differences in gene expression among groups.

## Results and Discussion

Diffusion maps extract latent variables that represent non-linear composites of genes that vary between samples (**Figure \@ref(fig:DMdm)**). 
We refer to these latent variables as diffusion components, the number of which is constrained by the number of samples in the data. 
Within each diffusion component (DC), a sample is represented by a single value and samples that have similar gene expression profiles among the genes captured in that component will have similar values.
Moreover, samples at the origin of a DC (i.e., near 0) have gene expression profiles that do not vary along that component, while samples with positive or negative values diverge. 
Each DC captures diminishing structure among samples with the first diffusion component (DC1) accounting for the largest variation among all samples. 
Differential expression analysis can then be used to identify genes that vary among the samples along a diffusion component. 
We expect the information extracted from each DC in this way will provide important insight into gene expression patterns caused by the time in fermentation (e.g., Brix level) and vineyard site. 

```{r DMdm, echo = F, message = F, warning = F, fig.cap ="Schematic of diffusion maps. Diffusion maps discover the underlying manifold in gene expression data through non-linear dimensionality reduction. **A)** When applied to many genes expressed by many samples, diffusion maps extract features that represent combinations of genes that drive similarities and differences among samples. **B)** These features are termed diffusion components. Samples at either extremum of the diffusion component are the most different from each other, while samples that fall at the origin are invariant along that component. In the above graphic, the orange and purple dots are the most different from one another, while the purple and green dot are most similar. Diffusion maps do not provide information on which genes lead to separation of samples along each diffusion component. **C)** Performing differential expression using the diffusion component as a continuous variable reveals the genes that significantly contribute to separation of samples. Gene gene1 has significantly higher expression in samples that fall on the right extrema of the diffusion component than samples that fall on the left extrema. **D)** While all genes are used to perform differential expression, not all genes are differentially expressed along an individual diffusion component. Gene gene2 is not differentially expressed along this diffusion component."}
knitr::include_graphics("figure/dm_dm.pdf")
```

### Diffusion mapping captures the global shift of gene expression during primary fermentation

We performed 60 inoculated primary fermentations of genetically similar Pinot noir grapes grown in 15 vineyards in California and Oregon over two vintages at the UC Davis Teaching and Research Winery (**Figure \@ref(fig:fig1dm)A**).
We profiled each fermentation using time-course RNA sequencing data, with the aim of using *S. cerevisiae* gene expression as an indicator of similarities and differences across fermentations. 
In the 2017 vintage, we took samples approximately corresponding to early growth phase (16 hours), late growth phase (40 hours), stationary phase (64, 88 hours), and end of fermentation (112 hours) (**Figure \@ref(fig:fig1dm)B**). 
In the 2019 vintage, we shifted sampling to capture cellular adaptation after inoculation (2 and 6 hours) and reduced sampling later in fermentation (16, 64, and 112 hours). 
The initial grape musts varied in parameters like initial nitrogen, pH, malic acid, tartaric acid, non-*Saccharomyces* microbial profile, and elemental profile while the final wines differed in volatile profiles and sensory outcomes (see Chapter [3](#chp-site)) [@cantu2020investigating]. 
Even still, from these data we determined that *S. cerevisiae* had a consistent CGEP across fermentations from different vineyards and vintages (see Chapter [1](#chp-brix)). 
Given the variable inputs and outcomes, we expected that there would be differences in *S. cerevisiae* gene expression that may include genes known to impact the sensory outcome of wine [@cordente2012flavour], given the sensory differences described for wines from these vineyards [@cantu2020investigating].
One fact making site-specific differences hard to quantify was that fermentations progressed at different rates leading to asynchronous biological progression among samples with respect to sampling time (**Figure \@ref(fig:fig1dm)C**). 
To address this issue and gain insight into vineyard-specific factors altering fermentation outcomes, we applied diffusion mapping to extract DCs and performed differential gene expression analysis to identify associated genes. 

```{r fig1dm, echo = F, message = F, warning = F, fig.cap = "Characteristics of samples taken throughout wine fermentation. **A)** Map of vineyard site American Viticulture Areas. **B)** Primary fermentation sampling time points for 2017 and 2019 vintages. Times are shown in hours and are relative to inoculation. **C)** Fermentation kinetics where brix (total soluble solids) is a proxy for remaining sugar. Fermentations proceeded at different rates."}
knitr::include_graphics("figure/dm_fig1.pdf") 
```

In the first DC, which accounts for the largest variation among all samples, we captured the dominant metabolic transition of *S. cerevisiae* during fermentation (**Figure \@ref(fig:fig2dm)A-B**). 
This is observed by the ordering of samples across DC1 in both the 2017 and 2019 vintages based on Brix. 
In a previous investigation, we used Brix to perform continuous differential expression and identified the CGEP of *S. cerevisiae* in Pinot noir fermentations (see Chapter [1](#chp-brix)).
To test whether diffusion mapping and DC1 captured the CGEP during fermentation, we performed differential expression over DC1 and compared this to log~2~ fold change values obtained for each gene previously tested by differential expression across the Brix variable (see Chapter [1](#chp-brix)).
Log2 fold change was strongly correlated between both methods of differential expression (**Figure \@ref(fig:fig2dm)C**, **Figure \@ref(fig:upsetgrid)**), indicating that DC1 captured the dominant global shift in gene expression during both 2017 and 2019 fermentations.

```{r fig2dm, echo = F, warning = F, message = F, fig.cap = "Diffusion mapping captures the core gene expression program of S. cerevisiae during wine fermentation. **A, B)** DC1 captures the metabolic transition that occurs as brix decreases during fermentation. The same transition occurs in the 2017 vintage and the 2019 vintage. Each point represents a sample from one time from one fermentation. Points that are closer along the x axis are more similar. The y axis is jittered to allow all points to be visualized. Points are colored by brix, a proxy for sugar concentration during fermentation, where brix = 0 indicates end of fermentation. **C)** High concordance between differentially expressed genes in diffusion component 1 and genes that are differentially expressed as brix decreases."}
knitr::include_graphics("figure/dm_fig2.pdf")
```

We next explored subsequent diffusion components (e.g., DC2, DC3, DC4, etc.) to determine other less dominant drivers of structure among samples. 
In general, within the 2019 data set we observed that lower diffusion components (DC2-DC4) captured structure across samples that was related to the time of fermentation (**Figure \@ref(fig:dc2dc4)**). 
For example, we observed transitions across fermentations that included cellular remodeling in early fermentation (DC2, DC3) and starvation during late fermentation (DC4) based on the differentially expressed genes associated with each diffusion component (**Figure \@ref(fig:dc2dc4)**). 
In contrast, higher diffusion components (DC5-DC8) showed patterns in the diffusion mapping data that suggest these DCs varied more based on site than on time in fermentation (**Figure \@ref(fig:dc5dc8)**). 
In addition, we saw that the number of differentially expressed genes diminished as the diffusion component number increased, indicating more subtle differences between samples (**Table  \@ref(tab:numdiffex)**). 
These patterns allowed us to determine specific differences between fermentations from individual vineyard sites.

```{r dc2dc4, echo = F, warning = F, message = F, fig.cap = "One-dimensional plots of DC2-DC4 for the 2019 vintage colored by **A)** AVA and **B)** hour. Diffusion components capture different latent structure in the dataset. Each component represents a different relationship among samples, but lower components capture shifts between stages of fermentation."}
knitr::include_graphics("figure/dm_dc2dc4.pdf")
```

```{r dc5dc8, echo = F, warning = F, message = F, fig.cap = "One-dimensional plots of DC5-DC8 for the 2019 vintage colored by **A)** AVA and **B)** hour. Lower diffusion components capture differences between fermentations within a stage of fermentation. Many samples from the same AVA cluster together along a single or multiple diffusion components, indicating similarity in gene expression profiles within an AVA."}
knitr::include_graphics("figure/dm_dc5dc8.pdf")
```

```{r numdiffex, echo = F, message = F, warning = F}
num_diffex <- "diffusion_component    |   induced    |   repressed    |   induced.1  |  repressed.1
DC1 | 335 | 435 | 470 | 457 
DC2 | 320 | 331 | 169 | 16 
DC3 | 122 | 127 | 32 | 293 
DC4 | 200 | 74 | 57 | 74 
DC5 | 80 | 481 | 13 | 0 
DC6 | 36 | 26 | 92 | 46 
DC7 | 82 | 56 | 50 | 7 
DC8 | 36 | 8 | 0 | 24"
df <- read.delim(textConnection(num_diffex), header=T, sep="|", strip.white=TRUE, stringsAsFactors=FALSE)
row.names(df)<-NULL
kable(df, "latex", booktabs = T, col.names = c("diffusion component", "induced", "repressed", "induced", "repressed"),
      caption = 'Number of significantly differentially expressed genes for the top diffusion components for the 2017 and 2019 vintages') %>%
  kable_styling(font_size = 9) %>%
  add_header_above(c(" " = 1, "2017" = 2, "2019" = 2), align = "l", escape = F)
```

Notably, while we detected the global shift in gene expression during primary fermentation from 2017 in DC1 (**Figure \@ref(fig:fig2dm)**), we did not observe clear delineations among higher DCs in the 2017 vintage (**Figure \@ref(fig:dm2017)**). 
All gene expression data are inherently noisy [@hansen2011sequencing], which is continuously improved upon based on rapid technical and methodological advancements in sequencing. 
Applying these advances, we sequenced the 2019 vintage with unique molecular identifier (UMI) barcodes and thus were better able to demultiplex sequencing lanes and remove PCR duplicates [@lohman2016evaluation].
With this improvement in data quality, we expect that signals associated with higher diffusion components were better captured in the 2019 sequencing data. 
These improvements were likely reflected in the number of significantly differentially expressed genes for DCs in 2017 and 2019 vintages (**Table  \@ref(tab:numdiffex)**). 

```{r dm2017, echo = F, warning = F, message = F, fig.cap = "One-dimensional plots of DC2-4 for the 2017 vintage colored by A) AVA and B) brix. Diffusion components capture different latent structure in the data set. Unlike in the 2019 vintage, samples in lower diffusion components do not cluster, potentially indicating a data quality issue that was remediated in the 2019 vintage through the use of UMI barcodes."}
knitr::include_graphics("figure/dm_dm2017.pdf")
```

### Diffusion mapping identified global shifts in gene expression during hypoxia

While diffusion maps successfully distinguished major fermentation transitions, we wanted to confirm that this method could faithfully report on other RNA-sequencing data where global shifts in gene expression occur. 
To address this, we identified a publicly available gene expression dataset of the onset of hypoxia in *S. cerevisiae* (GSE85595, GSE115171). 
Hypoxia occurs when a cell becomes oxygen limited, which is accompanied by large scale reprograming of gene expression for continued growth [@bendjilali2017time]. 
We applied diffusion mapping to this dataset and observed an ordered time-dependent transition to a hypoxic phenotype along DC1 (**Figure \@ref(fig:hypoxia)**). 
Sample positions indicated a rapid transition from 0-5 minutes of nitrogen exposure, indicating a fast metabolic transition to hypoxia that matured over the remainder of the time course. 
As part of this genetic reprogramming, a transient shift in gene expression has been previously identified at ~30 minutes of the hypoxic response and shown to partially overlap with the environmental stress response [@bendjilali2017time]. 
Within the diffusion mapping data, DC6 differentiated a transient shift at 30 minutes of hypoxia (**Figure \@ref(fig:hypoxia)**). 

```{r hypoxia, echo = F, warning = F, message = F, fig.cap = "Diffusion map of S. cerevisiae exposed to 100% nitrogen for 0-240 minutes. Only the trajectory of wild type samples are displayed. **A)** DC1 captured the transition from aerobic to anaerobic metabolism. **B)** DC6 captured a transient transcriptome remodeling at 30 min."}
knitr::include_graphics("figure/dm_hypoxia.pdf")
```

We then investigated whether genes differentially expressed along DC1 matched oxygen-regulated genes identified by previous studies of hypoxia. 
Across seven microarray studies, 11 genes (3 aerobic, 8 hypoxic) were consistently identified as being involved in aerobiosis or anaerobiosis in at least six studies (studies compiled by @bendjilali2017time).
We identified all 11 of these genes when considering a p < 0.05 for determining differentially expressed genes associated with DC1. We further compared our results to time-series RNA-sequencing profiles of wild type *S. cerevisiae* during hypoxic response [@bendjilali2017time].
Along DC1, 239 of 291 (82.1%) aerobic genes were significantly induced prior to exposure to nitrogen, while 422 of 519 (81.3%) hypoxic genes were significantly induced after prolonged exposure to nitrogen.
We found that genes induced at time zero were enriched in pathways like ribosome biogenesis, oxidative phosphorylation, and sterol metabolic process, while genes induced after prolonged exposure to nitrogen were enriched in pathways like oxidation reduction process, cell wall, glycogen metabolic process, and glycolysis/gluconeogenesis (**Figure \@ref(fig:DC1vizhyp)**). 
This matches our knowledge of the hypoxic transition in yeast [@kwast2002genomic; @bendjilali2017time]. 
These results indicate that diffusion mapping faithfully captured global changes in gene expression during the hypoxic shift, including transient states.

### Diffusion mapping detected metabolic remodeling throughout wine fermentation 

With the knowledge that diffusion mapping effectively identified gene expression program changes across two different data sets, we set out to further investigate the patterns captured within each DC. 
In the 2019 data, along DC2 there were clear separations amongst the 2, 6, and 16 hour samples, while the 64 and 112 hour samples fell on the origin.
Within the genes captured along this component, arginine biosynthetic process was enriched in genes that were more highly expressed in the 2 hour samples (*ARG1*, *ARG3*, *ARG5,6*, *ARG8*) (**Figure \@ref(fig:annotatedplts19hr)**, **Figure \@ref(fig:DC2viz2019)**). 
Arginine is the most abundant amino acid in Pinot noir grape must [@stines2000proline] and genes that encode proteins involved in arginine biosynthesis are suppressed by the presence of arginine [@crabeel1990arginine]. 
Expression of these biosynthetic genes in early fermentation likely reflects that *S. cerevisiae* is not completely adapted to the wine environment by 2 hours after inoculation. 
By 6 hours into fermentation, expression of these genes decreases, potentially signaling completion of cellular adaptation to the grape must environment. 
Interestingly, four of the 16 genes (YMR244W, YPR078C, YGL117W, YER085C) induced in 2 hour samples have no known function. 
Given that very few genes were induced at 2 hours and they were enriched for arginine biosynthesis, one speculation is that these genes may have functions related to nitrogen and arginine biosynthetic processes. 
Alternatively, expression of these genes may be associated with other cellular processes associated with adaptation to the wine environment.	

The 6 hours samples segregated to the opposite extrema of DC2, and thus were the most differentiated from the 2 hours samples along this component. 
Glycolysis was enriched among genes induced in these samples (**Figure \@ref(fig:DC2viz2019)**), which was also accompanied by gene expression changes that support a transition to anaerobic metabolism. 
For example, we detected induction of anaerobic translation elongation factor encoded by *ANB1* (**Figure \@ref(fig:annotatedplts19hr)**), which is optimally expressed below 0.5 umol/L O2 [@kwast1998oxygen], likely indicating a drop in must oxygen levels at this time. 
Genes important for cell wall processes were also induced at 6 hours, where *TIR1-4* were four of the top five genes induced (**Figure \@ref(fig:annotatedplts19hr)**). 
These genes encode cell wall mannoproteins that are required for anaerobic growth [@abramova2001reciprocal]. 
These genes are also important in DC3 for separating the 6 and 16 hours samples, along with many genes induced by anaerobiosis that included DAN1 and PAU genes (*PAU2*, *PAU3*, *PAU4*, *PAU5*, *PAU7*, *PAU8*, *PAU10*, *PAU11*, *PAU12*, *PAU15*, *PAU16*, *PAU17*, *PAU19*, *PAU20*, *PAU23*, *PAU24*) in the 16 hour samples [@sertil1997dan1; @rachidi2000saccharomyces]. 
Together, the induction of these genes regulated in response to oxygen across DC2 and DC3 likely signal the transition to anaerobiosis. 
In DC3 in the 16 hour samples, there were also many other biological processes, cellular compartments, and molecular functions enriched among the 293 genes that were induced (**Figure \@ref(fig:DC3viz2019)**), consistent with a transition to an active growth phase at this stage of fermentation. 
As diffusion components are ordered with those that capture the most variation among samples occurring first, DC2 and DC3 demonstrated that early metabolic remodeling was second only to gene expression changes that occur as brix decrease (e.g., entry into stationary phase) during fermentation.

Along DC4, we observed separation of the 64 hour samples from the 112 hour samples.
In the 64 hour samples, transmembrane transport, including amino acid and polyamine transport, were enriched categories among the genes that were induced (**Figure \@ref(fig:DC4viz2019)**). 
Many induced genes (*DUR3*, *DAL5*, *DAL7*) are involved in allantoin metabolism, a non-preferred nitrogen source (**Figure \@ref(fig:annotatedplts19hr)**). 
Induction of these genes at 64 hours likely indicates relief of nitrogen catabolite repression consistent with decreasing nitrogen concentrations and nutrient availability. 
Indeed, genes repressed by the presence of amino acids were induced in the 112 hours samples (*GAT2*, *ARG3*). 
Interestingly, *HXT13* and *MAN2* were among the top induced, genes along with *HXT17*, in the 112 hours samples (**Figure \@ref(fig:annotatedplts19hr)**). 
These *HXT* genes are mannitol transporters and *MAN2* encodes mannitol dehydrogenase [@Jordan2016]. 
These genes enable *S. cerevisiae* to metabolize mannitol as a non-preferred carbon source [@Quain1987; @Jordan2016; @ramakrishnan2016inter]. 
Mannitol is produced by non-*Saccharomyces* organisms, including fructophilic acetic acid bacteria like Lactobacillus kunkeii [@bisson2017two] and other non-*Saccharomyces* yeast [@mbuyane2018torulaspora]. 
Expression of these genes late in fermentation likely signals an increasingly generalized metabolic program that is utilizing non-preferred carbon sources as preferred sugars were exhausted. 

Overall, we interpret the patterns of separation along DC1-4 to be reflective of gene expression changes that occurred as *S. cerevisiae* proceeds through fermentation, adapts to the increasingly nutrient limited environment, and deals with associated stresses. 
While these changes appear common to the fermentations conducted here, future work will be required to address if individual processes captured in DC2-4 occur in the context of other wine strains and grape varieties or are unique to the wine yeast RC212 and Pinot noir fermentations. 
Nonetheless, these observations indicate that this analysis approach is a robust means for dealing with asynchronous gene expression data across fermentation. 
Moreover, it raises many questions about the genes important for defining separation along these DCs, which included gene products that are involved in arginine, mannitol, and anaerobic metabolism. 
Of notable interest are the large family of PAU genes, the vast majority of which have no known function in *S. cerevisiae*, but have been previously noted for being induced during fermentation and in response to stress [@luo2009functional]. 

### Lower diffusion components captured varied progression through fermentation 

Along lower diffusion components patterns of separation were based on the time of fermentation (discussed above), but we also observed samples from select vineyard sites that were outliers. We expect this indicates site-specific differences that influence *S. cerevisiae* activities during fermentation. 
For example, in the 2017 vintage, samples taken at 40 hours from SMV1, SMV2, and OR1 were shifted toward the 16 hour samples along DC1 indicating a different metabolic trajectory (**Figure \@ref(fig:fig2dm)**). 
Interestingly, we detected *Lactobacillus kunkeii* transcripts in primary fermentations from SMV sites (see Chapter [1](#chp-brix)).
L. kunkeii is a fructophilic lactic acid bacteria that produces mannitol and acetic acid during wine fermentation [@bisson2017two]. 
The observation of an altered metabolic program in early fermentation with the unique presence of *L. kunkeei* may indicate that inter-species interactions modulated *S. cerevisiae* gene expression.
In support of this, expression of genes involved in mannitol metabolism at 112 hours were the highest for the SMV sites (*HXT13*, *HXT17*, *MAN2*). 
One possibility is that *L. kunkeei* may have induced a [GAR+] prion state via acetic acid production, which is known to alter the metabolic strategy employed by *S. cerevisiae* [@jarosz2014cross] (see Chapter [1](#chp-brix)).
Following these observations in 2017, we attempted to detect the [GAR+] prion state in yeast isolated from the 2019 fermentations, but failed to detect [GAR+] yeast (see Chapter [1](#chp-brix)). 

Other outliers included SRH1, which in the 2019 vintage at 6 hours, and to a lesser extent at 2 hours, did not cluster with other samples from these time points along DC4 and DC6 (**Figure \@ref(fig:fig2dm)**, **Figure \@ref(fig:dc5dc8)**). 
The SRH1 6 hour samples were instead shifted toward other 16 hour samples, potentially indicating faster cellular adaptation to the wine environment. 
In support of this, fermentations from SRH1 initially decreased in Brix faster than other sites in both the 2017 and 2019 vintages (**Figure \@ref(fig:fig1dm)**). 
Another example along DC4 involved 64 hour samples from OR1 and OR2 being shifted toward the 112 hour cluster (**Figure \@ref(fig:fig2dm)**), indicating attenuated expression of genes induced in comparison to other 64 hour samples. 
Similarly, 112 hour samples from SNC1 and AS2 were shifted toward the 64 hour cluster (**Figure \@ref(fig:fig2dm)**). 
We expect these changes in the lower DC components that display strong patterns related to the time in fermentation likely reflect varied progression through fermentation due to nutrient differences. 

### Higher diffusion components identified vineyard site specific gene expression patterns

The common patterns, and existence of outliers, across lower diffusion components that aligned with fermentation progression and presence of non-*Saccharomyces* species indicated that information about specific sites may have been captured by these analyses. 
Given that some higher diffusion components clearly separated samples within a stage of fermentation, we used this information to determine vineyard-specific differences within those stages. 
In this way, we identified vineyard-specific gene expression patterns that can be used to infer *S. cerevisiae* activities specific to vineyard site(s).

To accomplish this, we concentrated on the 2019 samples that separated to the extremes within higher DCs as this separation indicated that these samples were the most differentiated at the transcriptome level. 
For example, samples taken at 2 hours separated along DC5 with SMV1, SMV2, SRH1, AV2, and RRV3 separating to one extrema and samples from RRV2, CRN1, SNC1, and AS2 separating to the other (**Figure \@ref(fig:dc5dc8)**). 
When comparing these sites, a standout difference was the induction of 13 genes in samples from SMV1, SMV2, SRH1, AV2, and RRV3 that were enriched for vitamin metabolic and cell wall processes (**Figure \@ref(fig:DC5viz2019)**). 
Given that co-culture experiments of wine fermentations have demonstrated that *S. cerevisiae* induces genes involved in cell wall remodeling and vitamin biosynthesis in response to the presence of non-*Saccharomyces* yeasts [@barbosa2015genomic; @kosel2017influence; @curiel2017different], we tested whether detected gene expression by non-*Saccharomyces* yeasts in 2 hour samples correlated with DC5. 
Indeed, DC5 correlated with total gene expression of *Hanseniaspora uvarum* (R2 = 0.49, p < 0.001), but not with total gene expression of other organisms (**Table \@ref(tab:nonsacc)**), suggesting that *H. uvarum* may be active in these early fermentations and impact *S. cerevisiae* metabolism. 
This is consistent with a previous study that found that *S. cerevisiae* remodels its cell wall in the presence of *H. uvarum* at three hours post-inoculation in wine fermentation [@curiel2017different]. 
Interestingly, PDC5 was among genes induced in fermentations with higher gene expression by *H. uvarum* (**Figure \@ref(fig:annotatedplts19ava)**). 
PDC5 encodes one of three isoforms of pyruvate decarboxylase, an enzyme involved in the formation of flavor–active higher alcohols in wine via the Ehrlich pathway [@terSchure2000].
This suggests that the presence of *H. uvarum* may lead to gene expression changes that directly impact wine sensory outcomes. 
The activity of this non-*Saccharomyces* yeast may also relate to the altered metabolic program detected in SMV1/2 and SRH1 fermentations within the lower diffusion components (see above). 
Interestingly, in a previous study we found that all fermentations had detectable *H. uvarum* DNA, but only a subset had detectable *H. uvarum* gene expression (see Chapter [3](#chp-site)).
Given the potential flavor active role of *H. uvarum*, in the future it will be important to determine the factors that lead to *H. uvarum* (in)activity in early fermentation.

```{r nonsacc, echo = F, warning = F, message = F}
nonsacc <- " organism | R2 | pvalue 
\\textit{Aureobasidium pullulans} | -0.03555 | 0.947
\\textit{Botrytis cinerea} | -0.03261 | 0.774
\\textit{Cladosporium sp SL 16} | -0.03341 | 0.804
\\textit{Hanseniaspora opuntiae} | -0.03524 | 0.911
\\textit{Hanseniaspora uvarum} | 0.490605 | < 0.001
\\textit{Lachancea thermotolerans} | -0.02741 | 0.638
\\textit{Metschnikowia fructicola} | 0.069637 | 0.086
\\textit{Pichia kudriavzevii} | -0.02819 | 0.654
\\textit{Rhizopus stolonifer} | -0.02439 | 0.582"
df <- read.delim(textConnection(nonsacc), header=T, sep="|", strip.white=TRUE, stringsAsFactors=FALSE)
row.names(df)<-NULL
kable(df, "latex", booktabs = T, col.names = c("organism", "R\\textsuperscript{2}", "p value"),
      caption = 'Correlation between total gene expression by non-Saccharomyces organisms and DC5. Only \\textit{Hanseniapsora uvarum} was significantly correlated with DC5.') %>%
  kable_styling(font_size = 9)
```

While samples from the SMV sites and SRH1 groups together at 2 hours, we detected separation of these fermentations at 16 hours along DC7 (**Figure \@ref(fig:dc5dc8)**).
This suggests that while these sites were initially similar, they differed later in fermentation. SMV and SRH are neighboring American Viticultural Areas in Southern California (**Figure \@ref(fig:fig1dm)A**). 
While few genes were induced in SMV samples, *ADH4* was the top induced gene (**Figure \@ref(fig:annotatedplts19ava)**). 
The *ADH* genes encode alcohol dehydrogenases that play an important role in fermentation by facilitating transitions between acetaldehyde and ethanol involving the redox cofactor NAD+. 
*ADH1* encodes the primary alcohol dehydrogenase isoform responsible for this reaction during wine fermentation [@ciriacy1975genetics]. 
Alcohol dehydrogenases are also involved in the formation of fusel alcohols within the Ehrlich pathway [@dickinson2003catabolism]. 
As such, differences in *ADH4* gene expression could be an important site-specific difference that has a role in wine aroma development. 
Other induced genes in SMV sites were involved in cell growth processes, including translation (*MRP2* and *TIF2*), transcription (*MED1*), and cell division (*CLB6*) (**Figure \@ref(fig:annotatedplts19ava)**). 
In site SRH1, induced genes along DC7 were involved in oxidative stress (*RCK1*) and sporulation (*SPO74* and *SSP1*), which may indicate a more challenging fermentation environment leading to altered gene expression patterns as compared to SMV sites at 16 hours (**Figure \@ref(fig:annotatedplts19ava)**). 
Given that genes associated with the Ehrlich pathway and fusel alcohol anabolism differentiated sites at 2 and 16 hours of fermentation, the Ehrlich pathway may be an important determinant of site-specific differences in Pinot noir wines. 

In the case of DC6 at 64 hours, OR1, OR2, and RRV2 samples segregated to one extrema (**Figure \@ref(fig:dc5dc8)**).
Interestingly, genes induced in these samples were associated with nitrogen limitation (*DAL5*, *PUT1*, *PUT2*; [@rossignol2003genome]), while genes involved in ammonia metabolism (*MEP3*, *SSY1*, *AUA1*) were induced within fermentations from sites at the other extrema. 
In line with these patterns reflecting differences in nitrogen, DC6 values correlated with initial grape must NOPA and NH~3~ measurements (initial NOPA: R2 = 0.62, p < 0.001, initial NH~3~: R2 = 0.60, p < 0.001), driven by low initial nitrogen in OR1, OR2, and RRV2 (**Figure \@ref(fig:DC6nitrogen)**). 
Interestingly, while initial nitrogen in OR1, OR2, and RRV2 was the lowest among all fermentations, these fermentations were supplemented approximately 24 hours after inoculation with diammonium phosphate (DAP) to adjust yeast assimilable nitrogen (YAN) levels to 250 mg/L.
Yet, these data indicate persistent nitrogen limitation for these sites, suggesting that a DAP addition at this timepoint may not be sufficient to meet nutrient requirements for some fermentations. 
DAP additions are common in commercial vinification, and these findings suggest that more research is needed to understand the impact of nitrogen additions on fermentation.

```{r DC6nitrogen, echo = F, message = F, warning = F, fig.cap = "DC6 separates samples by initial nitrogen concentration in the grape must. **A)** Initial concentration of NH~3~ and NOPA was lower in fermentations from OR1, OR2, and RRV2 than other vineyards. **B, C)** Initial NH~3~ and NOPA in grape must correlates with DC6, driven by low nitrogen in fermentations from vineyard sites OR1, OR2, RRV3."}
knitr::include_graphics("figure/dm_DC6nitrogen-1.pdf")
```

DC8 separated 112 hour samples with SNC1 and AS2 segregating from other samples (**Figure \@ref(fig:dc2dc4)**).
Interestingly, 14 of the 24 genes induced in these samples are of unknown function. 
Among these genes were *DDR2* and *HSP30*, which are stress related genes that are transcribed in response to a variety of environmental or physiological stresses [@kobayashi1996structure], as well as YDL218W that is induced in response to the mycotoxin patulin produced by a variety of molds [@iwahashi2006mechanisms].  
Associated with these stress-related genes were genes that function in meiosis and sexual reproduction, including *SPO74*, *MFA1*, and *AFB1*. 
These data are suggestive of stresses in these fermentations that could be driving the diploid wine yeast into a sexual reproduction cycle. 
This is of particular note, since the stresses associated with a wine fermentation environment are thought to impart strong selective pressures that drive adaptive evolution [@conacher2019peer]. 
This is in part reflected by the fact that *S. cerevisiae* strains associated with wine show a propensity for genetic diversity, including many instances of hybridization [@marsit2015diversity]. 
In future research, measuring metabolite concentrations throughout fermentation (anthocyanins, amino acids, oxygen) may reveal whether the same or different stress conditions lead to these patterns of gene expression.

Finally, across diffusion components samples from some fermentations from the same American Viticultural Area grouped together (**Figure \@ref(fig:dc2dc4)**, **Figure \@ref(fig:dc5dc8)**). 
For example, fermentations from OR, AV, and SMV sites often grouped together, providing support for the concept of American Viticultural Areas and regional terroir. 
However, we did not observe grouping among all fermentations from the same American Viticultural Area along all diffusion components. 
For example, samples from AS grouped together along DC5 (2 hours) and DC6 (64 hours), but not in DC8 (112 hours). 
Interestingly, the AS sites are separated by 1 km and yet separation along DC8 suggests there was detectable variation in *S. cerevisiae* metabolism in primary fermentation (**Figure \@ref(fig:dc2dc4)**, **Figure \@ref(fig:dc5dc8)**). 
Similarly, fermentations from RRV did not cluster together along any diffusion component, suggesting sub-appellations within the Russian River Valley were associated with significantly different *S. cerevisiae* gene expression patterns (**Figure \@ref(fig:dc2dc4)**, **Figure \@ref(fig:dc5dc8)**). 
The gene expression differences we detected across diffusion components provide candidate pathways that may underlie vineyard-specific outcomes in these sites. 

## Conclusion

In this study, we paired diffusion mapping with differential expression and captured global shifts in gene expression. This method revealed differences in primary fermentation of Pinot noir wine from 15 vineyard sites, as well as changes induced by hypoxia. 
Diffusion mapping was especially well suited for these studies as in both cases cells progressed asynchronously through transcriptome shifts with respect to sampling time. 
Through our analysis of wine fermentations, we found vineyard-site specific gene expression patterns that correlated with *H. uvarum* metabolic activity and initial nitrogen composition of grape must, as well as indications of sexual reproduction in select fermentations. 

Together, we expect these data provide important insight into the wine fermentation environment, including pathways, genes, and environmental factors that should be considered in the context of differential fermentation outcomes. 
Given the tremendous complexity of gene-environment interactions, we expect these data also serve to highlight the large amount of work to be done to understand both the biological mechanisms at play and how this knowledge can be applied by industry. 
In particular, we report transcriptomic heterogeneity that arises from the same strain of yeast, fermented in the same facility, using grape must from genetically identical plants. 
How this variability changes across the diverse landscape of wine yeasts strains and fermentation environments (e.g., grape varieties and associated chemical and microbiological profiles) remains to be seen. 
Importantly, we expect that the approaches pioneered here for studying *S. cerevisiae* gene expression in a complex environment using diffusion mapping provides an effective tool to probe these questions. 

## Methods

### Sampling, sequencing and preprocessing of wine fermentation samples

The winemaking protocol has been described previously [@cantu2020investigating; @grainger2020vineyard], but the relevant parts are reproduced with some added details below. 
The grapes used in this study originated from 15 vineyards in eight American Viticultural Areas in California and Oregon, U.S.A. 
All grapes were *Vitis vinifera* L. cv. Pinot noir clone 667, while was either rootstock 101-14 (AV1, RRV1, SNC1, SNC2, CRN1, AS1, AS2, SMV1, SMV2, SRH1), Riparia Gloire (OR1, OR2), or 3309C (AV2, RRV2, RRV3). 
We hand-harvested grapes at approximately 24 brix and transported the fruit clusters to University of California, Davis Teaching & Research Winery for fermentation. 
We performed separate fermentations for grapes from each site, with two fermentations for each site per vintage. 
We separated the hand-harvested fruit into half-ton macrobins on harvest day and added Inodose SO~2~ to 40 ppm. 
We then moved the bins into a 14C cold room until we destemmed the fruit and divided it into temperature jacket controlled tanks. 
We performed N~2~ sparging of the tank headspace prior to fermentation, and tanks were sealed with a rubber gasket. 
We performed a cold soak at 7C for three days and adjusted TSO~2~ to 40 ppm on the second day.
After three days, we increased the fermentation temperature to 21C and programmed pumpover frequency to hold the tank at a constant temperature.
Then we rehydrated *S. cerevisiae* RC212 with Superstart Rouge at 20 g/hL and inoculated fermentations with 25 g/hL of yeast.
Approximately 24 hours after inoculation, we adjusted nitrogen content in fermentations using DAP using the formula `(target YAN - 35 mg/L - initial YAN)/2`, and Nutristart using 25 g/hL. 
We only adjusted nitrogen if target YAN was below 250 mg/L in the initial grape must.
Approximately 48 hours after inoculation, we increased the fermentation temperature to 27C and again added DAP using the formula `(target YAN - 35 mg/L - initial YAN)/2`.
We ran fermentations to completion, when brix < 0. We sampled fermentations for brix at 16 hour and then every twelve hours relative to inoculation and sampled fermentations for RNA at 2 hours, 6 hours, 16 hours, 64 hours, and 112 hours (2019 vintage) and 16 hours, 40 hours, 64 hours, 80 hours, and 112 hours (2017 vintage) relative to inoculation.
For RNA samples, we performed a ten minute pumpover prior to sampling each tank and sampled 12 mL of wine and centrifuged at 4000 rpm for 5 minutes.
We discarded the supernatant and froze the pellet in liquid nitrogen, and then stored samples at -80C until RNA extraction.

### RNA Extraction and Sequencing

We extracted RNA using the Quick RNA Fungal/Bacterial Miniprep kit including DNAsel column treatment (cat#R2014, Zymo Research).
We eluted samples in 30 uL of molecular grade water and used 20 uL for sequencing. 
We used 3'-Tag-Sequencing single-end sequencing (Lexogen QuantSeq). 
In the 2019 vintage, we used UMI barcodes. 
The University of California, Davis DNA Technologies Core performed all library preparation and sequencing. 
We preprocessed samples according to manufacturer recommendations. 
First, we hard-trimmed the first 12 base pairs from each read and removed Illumina TruSeq adapters and poly A tails. 
Next, we used STAR to align our reads against *S. cerevisiae* S288C genome (R64, GCF_000146045.2) with parameters `--outFilterType BySJout --outFilterMultimapNmax 20 --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --outFilterMismatchNmax 999 --outFilterMismatchNoverLmax 0.6 --alignIntronMin 20 --alignIntronMax 1000000 --alignMatesGapMax 1000000 --outSAMattributes NH HI NM MD --outSAMtype BAM SortedByCoordinate` [@dobin2013star]. 
For the 2019 vintage, we used UMI tools to deduplicate alignments [@smith2017umi]. 
We then quantified reads mapping to each open reading frame using htseq count [@anders2015htseq]. 

### Hypoxia data set

We downloaded gene expression count data using from GEO using accession numbers GSE85595 and GSE115171.

### Construction of Diffusion Maps

We built diffusion maps as implemented previously [@fahimipour2020mapping]. 
To build diffusion maps from wine fermentations samples, we used k = 10 nearest samples, while for hypoxia we used k = 20. 
We increased the k size for hypoxia given the larger number of samples (n = 146 in 2017 vintage, n = 150 in 2019 vintage, and n = 336 in hypoxia).
Prior to diffusion map construction, we filtered gene counts to non-mitochondrial mRNA and normalized read counts based on total number of reads per sample (library size).

### Differential expression

To determine which genes drove separation of samples along each component, we used differential expression to correlate each gene with diffusion component values. 
We used limma to fit a linear regression model to each gene [@limma2015]. 
As input to differential expression, we used raw sequencing counts and performed filtering and normalization with the limma package using the `calcNormFactors()` function [@limma2015]. 
Using this model, the log~2~ fold change is the slope of the line for each unit increase in the diffusion component.
We normalized log~2~ fold change values by calculating the length of the diffusion component and multiplying all log~2~ fold change values by this amount, `(max - minimum) * log~2~ fold change`.
We analyzed log~2~ fold change values greater than two, i.e. genes with a log~2~ fold change of at least 2 between the most separated samples along a diffusion component. 
We performed Gene Ontology and KEGG enrichment analysis using the R clusterProfiler package [@yu2012clusterprofiler].

### Author Contributions

This manuscript was a collaboration by Taylor Reiter, Rachel Montpetit, Ron Runnebaum, C. Titus Brown, and Ben Montpetit.

|Author|Contributions|
|---|---|
|TER| Conceptualization; Data Collection; Formal Analysis; Methodology; Visualization; Writing - Original Draft; Writing - Review and Editing |
|RM | Data Collection |
|RR | Conceptualization; Funding Acquisition |
|CTB | Supervision; Funding Acquisition |
|BM | Conceptualization; Funding Acquisition |


```{r upsetgrid, message = F, warning = F, echo = F, fig.cap="Upset plot depicting intersections between differentially expressed genes using diffusion component 1 or brix as continuous variable. **A.** Intersection of differentially expressed genes from the 2017 vintage. **B.** Intersection of genes from the 2019 vintage. **C.** Intersection of genes differentially expressed along diffusion component 1 from the 2017 and 2019 vintages. **D** Intersection of genes differentially expressed relative to change in brix from the 2017 and 2019 vintages."}
knitr::include_graphics("figure/dm_upsetgrid-1.pdf")
```

```{r DC1vizhyp, echo = F, message = F, warning = F, fig.cap="Gene set enrichment for differentially expressed genes along DC1 during onset of hypoxia. All enriched categories with p < 0.05 after Bonferroni correction are shown. Pathways on the right side of the figure are induced in samples with a high diffusion component value, while pathways on the left of the figure are induced in samples with a low diffusion component value."}
knitr::include_graphics("figure/dm_DC1vizhyp-1.pdf")
```

```{r annotatedplts19hr, echo =F, message = F, warning = F, fig.cap=" Top genes induced along diffusion components that separate stages of fermentation in the 2019 vintage. Colored by hour at which samples were taken."}
knitr::include_graphics("figure/dm_annotatedplts19hr-1.pdf")
```

```{r DC2viz2019, echo = F, message = F, warning = F, fig.cap= "Gene set enrichment for differentially expressed genes along DC2 in the 2019 vintage. All enriched categories with p < 0.05 after Bonferroni correction are shown. Pathways on the right side of the figure are induced in samples with a high diffusion component value, while pathways on the left of the figure are induced in samples with a low diffusion component value."}
knitr::include_graphics("figure/dm_DC2viz2019-1.pdf")
```

```{r DC3viz2019, echo = F, message = F, warning = F,  fig.cap= "Gene set enrichment for differentially expressed genes along DC3 in the 2019 vintage. All enriched categories with p < 0.05 after Bonferroni correction are shown. Pathways on the right side of the figure are induced in samples with a high diffusion component value, while pathways on the left of the figure are induced in samples with a low diffusion component value."}
knitr::include_graphics("figure/dm_DC3viz2019-1.pdf")
```

```{r DC4viz2019, echo = F, message = F, warning = F, fig.cap="Gene set enrichment for differentially expressed genes along diffusion component 4 in the 2019 vintage. All enriched categories with p < 0.05 after Bonferroni correction are shown. Pathways on the right side of the figure are induced in samples with a high diffusion component value, while pathways on the left of the figure are induced in samples with a low diffusion component value."}
knitr::include_graphics("figure/dm_DC4viz2019-1.pdf")
```

```{r DC5viz2019, echo = F, message = F, warning = F, fig.cap="Gene set enrichment for differentially expressed genes along DC5 in the 2019 vintage. All enriched categories with p < 0.05 after Bonferroni correction are shown. Pathways on the right side of the figure are induced in samples with a high diffusion component value, while pathways on the left of the figure are induced in samples with a low diffusion component value."}
knitr::include_graphics("figure/dm_DC5viz2019-1.pdf")
```

```{r annotatedplts19ava, echo =F, message = F, warning = F, fig.cap="Top genes induced along diffusion components that separate fermentations within a stage of fermentation in the 2019 vintage. Colored by American Viticulture Area (AVA) of each sample."}
knitr::include_graphics("figure/dm_annotatedplts19ava-1.pdf")
```