# Transcriptomics provides a genetic signature of vineyard site with insight into vintage-independent regional wine characteristics {#chp-site}

```{=latex}
\begin{epigraphs}
  \qitem{We are enriched by our reciprocate differences.}%
        {Paul Valery}
\end{epigraphs}
```

```{r libs, echo = F, message = F, warning = F}
```

## Introduction

During vinification, grape must is transformed to wine through microbial metabolism, including fermentation of grape sugars into alcohols. In both inoculated and spontaneous fermentations, *Saccharomyces cerevisiae* often becomes the dominant fermentative organism due to a milieu of adaptations that support the rapid consumption of sugars and production of ethanol [@querol2003adaptive]. 
However, complex microbial communities consisting of other eukaryotic microorganisms and bacteria are present, active, and make significant contributions to the wine making process and final product [@jolly2003occurrence; @ghosh2015assessment; @wang2015viable; @bagheri2015diversity; @bagheri2020ecological].
Organisms often originate from the vineyard or the winery itself [@abdo2020establishment; @liu2020fungal]. 
In recognition of the important role these microbes have in wine making, non-*Saccharomyces* yeasts are increasingly inoculated into commercial fermentations due to the beneficial properties they can impart (e.g. bio-protection, lower ethanol, or distinct sensory characteristics, reviewed in @roudil2020non). 
Grape must treatment with sulfur dioxide (SO~2~) is also commonly used to control microbial populations and the growth of spoilage organisms, but many microorganisms survive SO~2~ treatment and contribute to fermentation outcomes [@egli1998dynamics; @bartowsky2009bacterial; @bagheri2020ecological]. 

The persistence of vineyard and winery derived microorganisms throughout the winemaking process has led to the idea that select microorganisms unique to a region or vineyard site may contribute to region-specific wine characteristics -- i.e. a component of *terroir* that is part of the natural environment that gives rise to wine -- by impacting wine fermentation outcomes (reviewed in  [@liu2019vineyard]).
In support of a role of microbial biogeography in wine terroir, microorganisms in vineyards, wineries, and grape musts are known to be associated with their region of origin [@bokulich2014microbial; @pinto2015wine; @wang2015fungal; @garofalo2016non; @bokulich2016associations; @mezzasalma2017grape; @mezzasalma2018geographical; @singh2018genotype; @vitulo2019bark; @liu2020fungal; @abdo2020establishment]. 
Moreover, the abundance of some organisms in grape must correlates with metabolite concentrations in finished wine, further linking microbial biogeography to fermentation outcomes and wine quality [@bokulich2016associations; @knight2015regional]. 

Still, relatively little is known about the influence of non-*Saccharomyces* microorganisms on wine fermentation outcomes, but studies are increasingly tackling this complex problem (reviewed in @conacher2019peer and @bordet2020yeast). 
Recent studies have documented increased glycerol accumulation and aroma profiles using sequential- or co-inoculation of *S. cerevisiae* with a single non-*Saccharomyces* yeast species under enological conditions [@sadoudi2012yeast; @renault2015increase; @englezos2016aroma; @lencioni2016controlled; @renault2016enhanced; @sadoudi2017metschnikowia; @gobert2017non; @whitener2017effect; @englezos2019saccharomyces; @binati2020contribution]. 
While outcomes are diverse, which may be expected given the variety of starting must and culture conditions used, many studies report consistent alterations in wine such as a higher glycerol content from fermentations inoculated with *S. cerevisiae* and *Starmerella bacillaris* [@englezos2016aroma; @englezos2019saccharomyces; @binati2020contribution]. 

How these altered fermentation outcomes occur remains a difficult question to address, as a given outcome may be the direct result of metabolism by the non-*Saccharomyces* organism, or the result of the non-*Saccharomyces* organism altering *S. cerevisiae* metabolism via direct or indirect interactions [@brou2018mixed; @curiel2017different; @alonso2019dominance]. 
In support of the latter, the presence of non-*Saccharomyces* organisms has been shown to increase the rate and diversity of resource uptake by *S. cerevisiae* in early fermentation [@tronchoni2017early; @curiel2017different; @alonso2019dominance]. 
In controlled steady-state bioreactor fermentations, the presence of *Lachancea thermotolerans* was found to increase the expression of *S. cerevisiae* genes important for iron and copper acquisition  [@shekhawat2019rna]. 
Such interactions are not limited to fungi -- lactic acid bacteria can induce epigenetic changes (e.g. [GAR+] prion) in *S. cerevisiae* that alter glucose metabolism [@brown2009heritable; @jarosz2014cross; @bisson2017two]
Such abilities of non-*Saccharomyces* organisms to impact *S. cerevisiae* metabolism and fermentation outcomes raises the question of whether microbial biogeography of vineyard sites persists in fermentations, thereby influencing wine outcomes in a site-specific manner. 
In addition, microbial diversity changes as primary fermentation progresses and *S. cerevisiae* becomes dominant [@morrison2018vineyard], suggesting a changing microbial community could feedback to impact fermentation progression in multiple distinct ways. 
Currently, we know relatively little about these inter-species interactions and how this influences *S. cerevisiae* fermentation performance, which as a field must be addressed if we are to understand how microbial community dynamics impact wine fermentation outcomes. 

Past inquiries into the microbial communities of grape must and wine related to regional distinctiveness have focused on assaying the presence of specific microbes based on ribosomal DNA amplicon sequencing [@bokulich2014microbial; @david2014high; @pinto2015wine; @wang2015fungal; @garofalo2016non; @bokulich2016associations; @mezzasalma2017grape; @mezzasalma2018geographical; @singh2018genotype; @liu2020fungal]. 
DNA sequencing has the advantages of capturing both metabolically active and inactive organisms, due to the relative stability of the DNA molecule, offering evidence of a rich history of the microbial community prior to sampling. 
Ribosomal DNA amplicon data further provides a measure of what microbes may be active at the time of sampling or may become active in the future. 
While microbiome DNA sequencing of grape musts supports regionally distinct microbial signatures, what microbes are metabolically contributing to fermentation outcomes remains largely unknown. 
This information is critical when considering the possibility that a particular microbe influences a wine fermentation outcome via metabolism or inter-species interactions. 

One measure of metabolic activity that is relatively accessible and can be applied at scale to address this issue is the measurement of gene expression in both *S. cerevisiae* and other non-*Saccharomyces* organisms. 
An interrogation of the genes that are "on" or "off" at a given time using RNA sequencing provides important information about the activities an organism is currently performing. 
In addition, the RNA molecule assessed by transcriptomics is constantly turned over within cells and is relatively unstable compared to DNA, which makes transcriptomics a reliable indicator of microbial activity at the time of sampling. 
For example, early in fermentation *S. cerevisiae* turns on genes required for glucose metabolism and represses expression of genes needed for the metabolism of other carbon sources; a pattern that flips towards the end of fermentation when glucose is depleted and *S. cerevisiae* must find alternative energy sources [@rossignol2003genome]. 
These patterns of gene expression are easily observed using transcriptomics [@rossignol2003genome; @marks2008dynamics], as such transcriptomics is increasingly being applied to understanding wine fermentation outcomes [@barbosa2015genomic; @tronchoni2017early; @curiel2017different; @alonso2019dominance; @shekhawat2019rna].

Here, we characterize microbial populations present in Pinot noir musts from California and Oregon in multiple vintages using both ribosomal DNA amplicon data from grape must samples and transcriptomics data from multiple fermentation time points. 
We demonstrate that genetic signatures of vineyard site and region are captured by these data. 
Total precipitation during growing season is one vineyard-associated factors that correlated with site-specific signatures. 
While DNA profiles reliably predict both vineyard site and region, these profiles did not correlate with the RNA profiles of the primary fermentations. 
This finding suggests other characteristics influence site-specific transcriptome signatures more than the grape must microbiome as measured by ribosomal DNA amplicon sequencing. 
Importantly, a comparison of DNA sequencing and transcriptomics data indicates that the majority of organisms detected by ribosomal DNA sequencing are not active during the primary fermentation, thus limiting the likelihood that many of these organisms significantly impact fermentation outcomes during the primary stage of fermentation.
Finally, using *S. cerevisiae* gene expression patterns and the function of genes identified, we are able to identify site specific factors that may be responsible for wine fermentation outcomes and wine sensory characteristics.
	
## Results and Discussion

To investigate associations between grape must microbial communities and regional distinctiveness of resulting wines, we performed standardized fermentations of Pinot noir wine using grapes from 15 vineyard sites in California and Oregon across multiple vintages (**Figure \@ref(fig:fig1)A**). 
In 2016, 2017, and 2019, we performed four inoculated fermentations per vineyard site using the wine yeast RC212, taking microbiome samples for DNA isolation and ribosomal DNA amplicon sequencing prior to inoculation. 
In the 2017 and 2019 vintages, we further profiled two primary fermentations from each site using RNA sequencing approaches to perform transcriptomics analyses at multiple fermentation time points. 
We performed all processing and fermentations at the UC Davis pilot winery to standardize vinification and minimize contributions from other factors (winery and winemaker) to the microbiome and transcriptome [@cantu2020investigating; @grainger2020vineyard].

```{r fig1, echo = F, message = F, warning = F, fig.cap = "Microbial diversity in grape must and fermentation microbiomes from different vineyard sites. **A)** Map displaying the 15 vineyard locations across eight American Viticulture Areas (AVAs) in California and Oregon. **B, C)** Relative abundance of taxonomic ranks in amplicon sequencing data capturing **B** Bacteria and **C** Fungi. Samples taken from fermentations from the same vineyard site and vintage are combined together and reflect relative abundance of organisms from four fermentation tanks. Only three tanks were fermented for AV2 in 2019 due to a smaller harvest. **D)** Relative abundance of all genes expressed by a detected organism during fermentation from the 2017 and 2019 vintages. The top plots shows all organisms and bottom plots displays only those organisms that account for less than 3% of mapped reads in each sample. Only organisms present in more than one fermentation are plotted."}
knitr::include_graphics("figure/sitefig1.pdf")
```

### DNA abundance by ribosomal amplicon sequencing is a poor predictor of detectable gene expression during fermentation

Using ribosomal DNA amplicon sequencing of bacteria and fungi, we detected 3254 distinct bacterial sequences and 2452 distinct fungal sequences in grape must (**Figure \@ref(fig:fig1)**), with mean species diversity per vineyard site higher for bacteria than for fungi (**Figure \@ref(fig:figs1)**). 
However, the core microbiome -- i.e. the species present in 90% of all grape musts across all vintages with at least 1% abundance – was larger for fungi than bacteria. 
The core microbiome consisted of 11 bacterial variants classified to nine taxonomic ranks and 19 fungal variants classified to 10 taxonomic ranks. 
All bacteria in the core microbiome belonged to the phylum *Proteobacteria* and were dominated by the genus *Tatumella* (**Figure \@ref(fig:figs2)**).
*Tatumella* has previously been identified as a dominant genera in other red wine fermentations where it correlated with total acid in grape must [@bubeck2020bacterial], however these associations have not been experimentally validated. 
Interestingly, three of the most abundant bacterial sequence variants identify to the acetic acid producing genus *Gluconobacter* (**Figure \@ref(fig:figs2)**). 
*Gluconobacter* is one of three genera of acetic acid bacteria associated with wine spoilage and the only genus we identify among dominant organisms (**Figure \@ref(fig:figs2)**) [@bartowsky2008acetic]. 
*Gluconobacter* are primarily active in grape must as the wine environment restricts growth of this genus [@bartowsky2008acetic].
Like bacteria, all fungi in the core microbiome belong to the phylum *Ascomycota*, however all fermentations were dominated by the genus *Hanseniaspora*, in particular *Hanseniaspora uvarum*. 
*H. uvarum* cannot complete alcoholic fermentation alone, but participates in and alters the quality outcomes of fermentation [@pietrafesa2020saccharomyces]. 
We also identified the fungal genus *Botrytis* among dominant organisms (**Figure \@ref(fig:figs2)**), although we lack the power to resolve whether the particular variant we detected belongs to the spoilage organism *Botrytis cinerea*.

```{r figs1, echo = F, message = F, warning = F, fig.cap = "Bacterial and fungal ribosomal DNA amplicon sequencing Chao 1 and Shannon alpha diversity for mean species diversity per vineyard site, averaged across vintages. **A** Bacteria. **B** Fungi."}
knitr::include_graphics("figure/sitefigS1.pdf")
```

```{r figs2, echo = F, message = F, warning = F, fig.cap = "Top 20 most abundant ribosomal DNA amplicon sequencing variants across vintages. Labelled as genus or the next lowest taxonomic rank of classification. **A** Bacteria. **B** Fungi. Tatumella was the most abundant bacterial amplicon sequencing variant across vineyards and vintages, while Hanseniaspora was the most abundant fungal amplicon sequencing variant."}
knitr::include_graphics("figure/sitefigS2.pdf")
```

Through this work, we have extended microbiome must sequencing to include the 2019 vintage, with results largely matching findings from previous vintages across these same vineyard sites (Steenwerth et al. 2020, *in review*). 
The observed microbial community composition is consistent with other vineyard site analyses and contained expected organisms known to be present at the initial stages of the wine making process. 
Ribosomal DNA amplicon sequencing of grape must captures both live and inactive/dead cells. 
In contrast, transcriptome profiling via RNA sequencing captures metabolically active cells and the genes expressed at the time of sampling. Using 3' Tag RNA sequencing (3' Tag-seq), we profiled eukaryotic organisms that were transcriptionally active during fermentation using samples taken at multiple fermentation time points (i.e. 16, 64, and 112 hours after inoculation in 2017 and 2, 6, 16, 64, and 112 hours after inoculation in 2019. We performed our analyses with the time points that matched between both vintages except where explicitly state.).
While traditional RNA-sequencing produces sequencing reads from an entire transcript, 3' Tag-seq produces one molecule per transcript by sequencing approximately 100 base pairs upstream of the 3'-end of a sequence [@lohman2016evaluation]. 
This sequencing chemistry requires a poly(A) tail, limiting the sequenced fraction of the transcriptome almost entirely to polyadenylated eukaryotic mRNAs.

From the resulting 3' Tag-seq data, we observed that relatively few eukaryotic microbes were transcriptionally active during fermentation (**Figure \@ref(fig:fig1)D**).
Considering all 15 sites together, only 18 eukaryotic species were detected to be transcriptionally active in these Pinot noir fermentations.
Further reflecting this finding, *S. cerevisiae* transcripts accounted for the majority of sequences across all fermentations at all time points. 
To assess whether non-inoculated *S. cerevisiae* strains were transcriptionally active during fermentation and in part responsible for some fraction of sequence reads, we compared each transcriptome against all annotated *S. cerevisiae* genomes in GenBank, as well as a genome assembly of *S. cerevisiae* RC212. 
While non-RC212 *S. cerevisiae* strains were detectable in every fermentation, this fraction accounted for less than 1% of uniquely identifiable sequences. 
This demonstrates that the inoculated RC212 strain dominated fermentations at all sampled time points. 
Interestingly, we also identified *Vitis vinifera* transcripts in all samples (**Figure \@ref(fig:fig1)D**). 
The presence of *V. vinifera* transcripts was not expected given general fermentation conditions and suggests intact and potentially active grape cells persist through out fermentation.

In comparing specific organisms detected via DNA sequencing and transcriptomics, we see that only 4 (*Aureobasidium pullulans*, *H. uvarum*, *Hanseniaspora vineae*, and *S. cerevisiae*) of 397 distinct fungal species definitively identified by ribosomal DNA profiling were likely active based on transcriptomics data. 
RNA sequencing samples were taken as early as 2 hours after inoculation, suggesting that organisms detected by amplicon sequencing had loss activity prior to or concurrent with inoculation; this is well before *S. cerevisiae* would begin to produce inhibitory concentrations of ethanol. 
Of the four detected organisms, *H. uvarum* was the most abundant in musts prior to inoculation based on amplicon sequencing and was detected in grape musts from every vineyard site (**Figure \@ref(fig:fig2)A**). 
Still, the relative abundance of DNA in grape must was only weakly correlated with relative abundance of RNA during fermentation (R^2^ = 0.14, p < 0.01). 
Importantly, while these values are correlated, *H. uvarum* had almost no detectable gene expression in fermentations from many sites where it dominated the DNA profile of the grape must (**Figure \@ref(fig:fig2)B**). 
Finally, even when we performed this analysis using samples from the first hours of fermentation after inoculation, relative abundance of *H. uvarum* DNA in grape musts remained weakly correlated with relative abundance of RNA (two hours: R^2^ = 0.21, p < 0.05, six hours: R^2^ = 0.28, p < 0.01). 
In the case of A. pullulans, DNA in grape must is not correlated with gene expression during fermentation (two hours: R^2^ = -0.03, p = 0.6, six hours: R^2^ = -0.025, p = 0.53). 
These results indicate that DNA profiling does not reflect metabolic activity of most eukaryotic microorganisms in inoculated fermentations even when the organisms are detected at high abundance and are detectable via both sequencing methods.

```{r fig2, echo = F, message = F, warning = F, fig.cap = "*H. uvarum* ribosomal DNA amplicon sequencing data does not meaningfully correlate with relative abundance in RNA sequencing data. **A)** Bar chart of relative abundance of *H. uvarum* compared to other non-Saccharomyces species across fermentations from each site based on amplicon sequencing data of ribosomal DNA. **B)** Scatter plots relative abundance of *H. uvarum* as determined by amplicon sequencing of ribosomal DNA (x-axis) vs. RNA sequencing (y-axis)."}
knitr::include_graphics("figure/sitefig2.pdf")
```

Given these findings, it is important to consider if a lack of gene expression data for non-*Saccharomyces* species could be reflective of some other issue that is technical or biological in nature. 
We consider this highly unlikely for two reasons. 
First, both DNA and RNA sequencing require similar protocols for extraction of nucleic acids from cells that should perform approximately equally across samples. 
Second, RNA sequencing relies on a highly conserved biological processes (mRNA polyadenylation), hence while we could envision RNA sequencing failing for one or a few organisms, it should not fail across many fermentations for the large majority of organisms seen in this work. 
Moreover, of the 16 non-*Saccharomyces* fungi detected via RNA-sequencing, eight of these organisms were not detected at the genus level by DNA profiling (*Cladosporium sp SL-16*, *L. thermotolerans*, *Metschnikowia fructicola*, *Metschnikowia sp. AWRI3582*, *Pichia kudriavzevii*, *Preussia sp. BSL10*, *Rhizopus stolonifer*, *S. bacillaris*).
This suggests that transcriptomic profiling is a sensitive assay able to detect active organisms present in a population that are missed by ribosomal DNA amplicon sequencing likely due to decreased resolution resulting in the inability to resolve genus or species. 
Notably, some of these organisms likely influence fermentation outcomes: in mixed fermentations with *S. cerevisiae*, *S. bacillaris* has been shown to lower the final ethanol concentration and increase the concentration of glycerol [@englezos2018specific], while M. fructicola increased the concentration of esters and terpenes [@boscaino2019impact].
Therefore, the detection of these organisms by transcriptomics provides valuable information with respect to the active microbial population in fermentations.
Our findings align well with another recent report that showed an RNA-based sequencing strategy is a highly sensitive alternative to amplicon sequencing [@cottier2018advantages]. 
As such, it may be appropriate to use RNA-sequencing as a general method to capture the metabolically active community during wine fermentation, especially when drawing a connection between the wine microbiome and fermentation outcomes.

## Genetic signatures differentiate vineyard site, region, and vintage

The region and site from which grapes are harvested can have an important influence over the character of a resulting wine based by on a variety of factors (e.g. climate, soil type, grape associated microbes etc.). 
As such, we considered if the data generated using DNA sequencing and RNA sequencing during fermentation reflects each of the 15 vineyard sites through unique genetic signature. 
To investigate this concept, we grouped DNA and RNA sequencing samples by vineyard site, region, and vintage to see if there were detectable differences among these groups. 
Using analysis of similarities (ANOSIM; see methods), we determined that all three factors explain differences among groups of samples, with vineyard site explaining the most  similarity of groups built with the DNA and RNA sequencing data (**Figure \@ref(fig:fig3p1)A-D**). 
This supports the idea that fermentations have a detectable genetic signature that is reflective of vineyard site. 
To understand which specific organisms and genes contribute to the genetic signatures of both vineyard site and region, we built machine learning classification models using random forests. 
These models weight the contribution of each feature to predictive accuracy of the model, enabling robust identification of specific genes or organisms that differentiate vineyard sites or regions among fermentations. 
When we used transcriptome data from all vintages in model training and testing to predict region, we achieved 100% accuracy, which decreased to 91% and 87% accuracy when using ribosomal DNA data for bacteria and fungi, respectively (**Figure \@ref(fig:tabs1)**; **Figure \@ref(fig:figs4)**; **Figure \@ref(fig:figs5)**). 
When we instead used data from one vintage in model training and testing to predict region, accuracy dropped across all models but ranged from 53%-78% (**Figure \@ref(fig:tabs1)**; **Figure \@ref(fig:figs4)**; **Figure \@ref(fig:figs5)**). 
Models built with fermentations from all vintages can better capture cross-vintage similarities, but may not generalize to future vintages. 
Conversely, models built on a subset of vintages will generalize to future fermentations but may not incorporate the best predictive variables.
Given this, we assumed that models trained using data from a single vintage better reflected model accuracy, but that models trained data from all vintages better reflected cross-vintage similarities. 
To best identify cross-vintage similarities, we analyzed cross-vintage models moving forward.

```{r fig3p1, echo = F, message = F, warning = F, fig.cap = "Genetic profiles correlate with vineyard, region, and vintage as well as some vineyard site and initial grape must characteristics. **A-C)** Non-metric Multi-dimensional Scaling plots of Aitchinson dissimilarity of **A** bacterial communities, **B** fungal communities, and C and transcriptomes across vintages. The closer two points are on the graph, the more similar their genetic profiles are. **D)** Vineyard site, region, and vintage account for genetic diversity patterns in Analysis of Similarity (ANOSIM). R values represent strength of association, with higher R values indicating stronger grouping according to the parameter. All values are significant (p < 0.001). **E)** Percent of accuracy attributable to different organisms in random forests models. A higher percentage of variable importance was attributable to S. cerevisiae and V. vinifera in models that predicted region than vineyard site. **F, G)** Correlograms representing similarities between fermentation metrics. **F** Grape must chemical parameters and vineyard site characteristics were correlated. Squares are labelled with correlation values from Pearson’s correlation. Only comparisons with p < 0.05 are displayed. **G** Bacterial, fungal, and transcriptome profiles correlated with some vineyard site and grape must chemical characteristics. Squares are labelled with correlation values from Mantel tests. Only comparisons with an FDR < 0.1 are displayed. PPT: precipitation, GDD: growing degree days, MA: malic acid, TA: titratable acidity."}
knitr::include_graphics("figure/sitefig3.pdf")
```

We were surprised to find that bacterial models outperformed fungal models for classification of region, as bacterial must samples added the least predictive power in previous models for region prediction [@bokulich2014microbial], including for Pinot noir grapes grown in Australia  [@liu2020fungal]. 
Bacterial must samples have also been shown to be predictive of region in Californian Chardonnay, but not Californian Cabernet Sauvignon [@bokulich2014microbial], suggesting a possible cultivar-specific effect. 
In previous inquiries, samples were processed in vineyard-specific wineries, providing another variable that could potentially alter the measured microbiomes and the contributions attributed to bacteria and fungi.

```{r tabs1, echo = F, message = F, warning = F, fig.cap = "Performance of random forests models on trained and validated on different sets of data for bacterial and fungal ribosomal DNA amplicon sequencind data and RNA sequencing data.  In the validation column, *30* refers to the accuracy of models trained with 70% of samples from all vintages and validated on the left out 30%."}
knitr::include_graphics("figure/sitetabS1.pdf")
```

When we used the same data to generate vineyard-specific models, predictive accuracy was on average 20.3% less than region-specific models (**Figure \@ref(fig:tabs1)**). 
However, it is important to note that this drop in accuracy was driven by within-region misclassification for vineyards in Willamette Valley (31 km separation), Santa Maria Valley (5 km separation), and Arroyo Seco (1 km separation) American Viticulture Areas (AVA) (**Figure \@ref(fig:figs3)**).
The same misclassifications persisted across many models, highlighting potential within-region similarity that contributes to genetic signatures, which fits well with the concept of AVA and region-associated wine characteristics. 
Given that random forests models estimate the importance of each gene in determining vineyard or region classification, we further used the transcriptome models to gain insight into biological differences between vineyard sites and regions.
For this, we calculated the percent of total importance attributable to each gene from each organism. In general, vineyard-specific models weighted non-*Saccharomyces* microbial genes as a whole as more important for predictive accuracy than region-specific models (**Figure \@ref(fig:fig3p1)E**). In particular, genes from *S. bacillaris*, *M. fructicola*, *Metschnikowia sp. AWRI3582*, and *L. thermotolerans* were important for vineyard site classification. 
The ability of non-*Saccharomyces* gene expression to distinguish site is likely related to the unique combination of non-*Saccharomyces* organisms present in each fermentation, which results in these organisms having strong predictive power when detected. Moreover, sites that are closer together may have more similar *S. cerevisiae* and *V. vinifera* gene expression since the fermented musts come from grapes grown in environments with similar biotic and abiotic conditions. 
This is idea is extended by the finding that regional models weight *S. cerevisiae* and *V. vinifera* genes as higher importance (**Figure \@ref(fig:fig3p1)E**), where sites separated by larger distances (i.e. in different regions) would experience more varied biotic and abiotic factors, which would be expected to impact the resulting must and overall fermentation. 
To more directly address how environmental factors and grape must chemistry correlate with genetic signatures, we correlated initial must chemical parameters (pH, titratable acidity, malic acid, NOPA, and NH~3~) and vineyard site characteristics (total precipitation, growing degree days, and geographic distance between sites) with DNA and RNA profiles using the Mantel test (see methods). 
From these analyses, we found geographic distance between vineyards correlates with precipitation and growing degree days, indicating that sites that are geographically closer experience more similar weather patterns, as would be expected (**Figure \@ref(fig:fig3p2)A**). Among the factors tested, only precipitation correlated with all genetic profiles (**Figure \@ref(fig:fig3p2)B**).

```{r fig3p2, echo = F, message = F, warning = F, fig.cap = "Correlograms representing similarities between fermentation metrics. **A** Grape must chemical parameters and vineyard site characteristics were correlated. Squares are labelled with correlation values from Pearson’s correlation. Only comparisons with p < 0.05 are displayed. **B** Bacterial, fungal, and transcriptome profiles correlated with some vineyard site and grape must chemical characteristics. Squares are labelled with correlation values from Mantel tests. Only comparisons with an FDR < 0.1 are displayed. PPT: precipitation, GDD: growing degree days, MA: malic acid, TA: titratable acidity."}
knitr::include_graphics("figure/sitefig3p2.pdf")
```

Similarly to geographic distance, initial chemical profiles of vineyard sites were more similar when sites are geographically closer, which may reflect why regional models weight *S. cerevisiae* and *V. vinifera* genes as higher importance as compared to vineyard-specific models.
However, we found surprisingly few correlates between genetic profiles and initial grape must conditions (**Figure \@ref(fig:fig3p2)B**). 
While fungal profiles correlate with initial malic acid, NOPA, and NH~3~ and bacterial profiles correlate with initial NOPA, transcriptome profiles only correlate with initial malic acid. 
Interestingly, transcriptome profiles do not correlate with initial nitrogen concentration even though nitrogen availability is often the trigger for *S. cerevisiae* to enter stationary phase during fermentation, thus remodeling expression of hundreds of genes [@rossignol2003genome]. 
However this may be specific to our experimental set up given that we back-added nitrogen to nitrogen-limited fermentations, as is common in industrial vinification. 
These results suggest that other unmeasured factors drive distinctiveness in these fermentations which we do not capture in these analyses.

### *S. cerevisiae* gene expression provides insight into vineyard site  and region features

*S. cerevisiae* is likely the best understood eukaryote based on the use of this organism as a model system for biology, which has provided a rich set of genomic resources and databases [@cherry1998sgd]. 
As such, *S. cerevisiae* gene expression can be used as a biosensor to provide insight into the fermentation environment based on activities yeast perform.
The utility of this data is furthered by the fact that *S. cerevisiae*  gene functions are well studied in the context of wine production, *S. cerevisiae* is ubiquitous across all fermentations, and the transcriptomics data is dominated by reads from *S. cerevisiae* (e.g. data completeness). 
Consequently, given the data above suggesting unknown factors are directing fermentation outcomes, we queried the *S. cerevisiae* transcriptomics data to assess what genes were important for predicting region and vineyard site to infer what may be unique about each vineyard site or region. 

From this analysis, important predictors of both site and region included flavor-associated genes  involved in the formation of higher alcohols and volatile fatty acids through the Ehrlich pathway (*BAT1*, *BAT2*, *ARO9*, *PDC1*, *PDC5*, *PDC6*, *ADH7*, *YPR1*). 
It is known that high expression of *BAT1* increases the formation of isoamyl alcohol, its acetate ester, and isobutanol, while expression of *BAT2* increases the formation of isobutanol and isobutyric acid [@lilly2006effect].
Genes involved in the formation volatile sulfur compounds were also important predictors of region (*MET25*, *BNA3*, *DUG1*, *PRC1*, *GAP1*) and site (*MET25*, *PRC1*, *GAP1*). 
Given that these genes were detectable as indicators of both region and site, site-variable expression of these genes could contribute to region- and vineyard-specific wine flavor profiles detected in wines from these vineyards in previous vintages [@cantu2020investigating]. 
At this time, it remains unknown what factors cause these flavor-associated genes to differ between fermentations.

In addition to flavor-associated genes, many *S. cerevisiae* genes that are important for predicting vineyard site were members of the Com2 regulon. 
Expression of these genes is protective against SO~2~ stress [@lage2019transcriptomic]. 
SO~2~ is used in vinification to reduce the growth of spoilage organisms that generate off-flavors. 
Wine strains of *S. cerevisiae* are more tolerant to SO~2~ stress than lab strains, but SO~2~ exposure causes inhibition of key metabolic enzymes like alcohol dehydrogenase, as well as other processes through cleavage of disulfide bonds [@hinze1986analysis; @divol2012surviving]. 
Of 511 genes with SO~2~ transcriptional activation that is dependent on Com2 [@lage2019transcriptomic], 192 were important for differentiating site in our predictive model. 
A subset of 162 of the Com2 regulon genes were also important for predicting vineyard region.
Within these gene lists are genes involved in the efflux of sulfite and bisulfite (*SSU1*, *ECM34*), sulfate assimilation (*SUL1*, *SUL2*), sulfate reduction (*MET3*), biosynthesis of methionine, cysteine, arginine, and lysine (*MET1*, *MET4*, *MET30*, *MET32*, *MTH1*, *LYS9*, *ARG3*, *ARG4*, *ARG7*), biosynthesis of the sulfur-containing vitamin biotin (*BIO3*, *BIO4*, *BIO5*, *ISA1*), assembly of the V-ATPase responsible for mitigating intracellular acidification (*VMA7*, *VMA11*), and inactivation of glycolysis (*ATP1*, *COX14*).
We treated all fermentations with an equal dose of SO~2~ at the beginning of vinification; however, variable application of sulfur-containing pesticides in the vineyard may lead to disparate SO~2~ stress during fermentation underlying the signatures of site and region that we observed. 
This suggests that SO~2~ stress response is an important factor that should be considered, especially in regions or vineyards with strong disease pressures that require application of sulfur-containing pesticides. 
This is important given that the expression of pathways related to sulfur metabolism can have a profound impact on the sensory attributes of wine [@cordente2012flavour].

```{r figs7, echo = F,warning=F, message = F, fig.cap = "Percent of accuracy attributable to different organisms in random forests models.  A) Importance of genes expressed by different organisms in the overall model. **A)** higher percentage of variable importance was attributable to S. cerevisiae and V. vinifera in models that predicted region than vineyard site **B)** Importance of genes expressed by different organisms for the six vineyard sites that had high classification accuracy using random forests models. Genes expressed by non-Saccharomyces organisms are more important for predicting vineyard sites RRV3, AV2, and SMV2, while genes from S. cerevisiae and V. vinifera are more important for predicting vineyard sites OR1, RRV1, and AS2."}
knitr::include_graphics("figure/sitefigS7.pdf")
```

To further explore these connections between *S. cerevisiae* gene expression and site, we identified the sites with the lowest rates of misclassification in models built with transcriptome data, since these sites would be expected to contain the strongest genetic signatures. These sites were OR1, RRV1, RRV3, AV2, AS2, and SMV2 (**Figure \@ref(fig:fig4)A**). 
Of these six sites, *S. cerevisiae* and *V. vinifera* drove predictive accuracy for OR1, RRV1, and AS2 (**Figure \@ref(fig:fig5)**), while other single microorganisms were most important for RRV3, AV2, and SMV2 (*A. pullulans* in RRV3, *L. thermotolerans* in AV2, and *S. bacillaris* in SMV2, see **Figure \@ref(fig:figs7)**). 
Within these six sites, we first looked at genes that were important for predicting each vineyard site, but the majority of genes were predictive for a single site (**Figure \@ref(fig:fig4)C**). 
This is notable, since it indicates that each of these sites had distinct *S. cerevisiae* gene expression patterns likely indicative of underlying differences in the fermentations. 
In support of this, Gene Ontology pathway enrichment analysis demonstrated gene sets from each site differ in cellular component and molecular function enrichments (**Figure \@ref(fig:fig4)B**). 
Interestingly, predictive genes from OR1 are enriched for nicotinate and nicotinamide metabolism, including many genes that are involved in the NAD+ salvage pathway (*PNC1*, *NMA2*, *NRK1*, *NPT1*). 
For glyceraldehyde-3-phosphate to progress through fermentation it requires NAD+ as an electron acceptor. 
Typically, NAD+ is regenerated from NADH in the last reaction of alcoholic fermentation, but if NAD+ is limited, it can also be regenerated by respiration or conversion of dihydroxyacetone to glycerol [@ansell1997two]. 
We see signatures of these processes across all fermentations as described in Chapter [1](#chp-brix), but only in fermentations from OR1 is the NAD+ salvage pathway also enriched. 
Fermentations from OR1 proceed relatively slower than those from other sites (**Figure \@ref(fig:figs8)A**), indicating that NAD+ availability may be a factor underlying slower fermentation kinetics. 

```{r fig4, echo = F, message = F, warning = F, fig.cap = "**A)** Percent of accuracy attributable to different organisms for predicting the six vineyard sites that had high classification accuracy using random forests models. Genes expressed by non-Saccharomyces organisms are more important for predicting vineyard sites RRV3, AV2, and SMV2, while genes from S. cerevisiae and V. vinifera are more important for predicting vineyard sites OR1, RRV1, and AS2. **B)** Gene Ontology and KEGG pathways that are enriched among the S. cerevisiae genes that are important for predicting each vineyard site. **C)** Upset plot depicting the intersection of genes expressed by S. cerevisiae that are important for predicting vineyard site for the subset of fermentations that had high predictive accuracy across random forests models. The majority of genes are important for predicting a single vineyard site. Only intersections with more than 10 genes are displayed. "}
knitr::include_graphics("figure/sitefig4.pdf")
```

Continuing to explore site-specific genes, the top most predictive gene expressed by *S. cerevisiae* for fermentations from AV2 was *FRE3*, while the twelfth most predictive gene was *FTH1*, both of which are genes involved in iron homeostasis and are induced by low iron levels.
One possible explanation for the importance of these genes in site predictive models is that wines made from AV2 had the second lowest iron concentration across all sites in the 2017 vintage and the third lowest in the 2019 vintage (**Figure \@ref(fig:figs8)B**). 
This  may suggest an iron deficiency in the must that would lead to expression of these genes to maintain iron homeostasis. 
Alternatively, in steady-state reactor fermentations it was shown that *S. cerevisiae* induced expression of genes associated with copper and iron uptake after 48 hours of co-culture with *L. thermotolerans*  [@shekhawat2019rna]. 
Given that genes expressed by *L. thermotolerans* contributed the largest fraction of importance amongst non-*Saccharomyces* species for predicting site AV2, it is also possible that the presence of *L. thermotolerans* was responsible impacting iron-associated gene expression through inter-species competition or interaction. 
These findings (e.g. presence of *L. thermotolerans*, low iron levels in wine made from AV2, and importance of genes associated with low iron conditions) suggest iron metabolism may be central to fermentation outcomes for grapes harvested from AV2. 
This is of particular interest because iron (Fe2+/Fe3+) is a catalytic redox cofactor both for reactions involving polyphenols and oxygen in wine [@danilewicz2003review], and in *S. cerevisiae* metabolism [@shakoury2004transcriptional]. 
As such, in combination with oxygen and other factors that contribute to redox balance, iron concentration may substantially impact wine outcomes both via wine chemistry and through *S. cerevisiae* metabolism.

```{r figs8, echo = F, message = F, warning = F, fig.cap = "**A)** Brix measurements throughout fermentation in the 2017 and 2019 vintages. Fermentations from OR1 are demarcated with shapes. Brix reflects the total soluble solids in a fermentation and is a proxy for sugar concentration in fermentations. Brix start high (0 hours, 25 brix) and decrease until the fermentation is dry (>100 hours, 0 brix). **B)** Iron concentration in final wines from different vineyards in the 2017 and 2019 vintages."}
knitr::include_graphics("figure/sitefigS8.pdf")
```

Interestingly, the gene with the most importance expressed by *S. cerevisiae* for predicting site RRV3 was *DAL4*, with *DAL5*, *DAL7*, and *DAL1* also having predictive power. 
The *DAL* genes encode proteins involved in degradation of the alternative nitrogen source allantoin. 
Expression of these genes is low across fermentations from RRV3 compared to that in other sites, potentially suggesting a lack of allantoin in the grape must. 
Allantoin accumulation enhances reactive oxygen species scavenging in response to abiotic stress in Arabidopsis and likely in *V. vinifera* [@you2017vitis].

Taken together, these results provide a rich list of genes that are directly linked to wine flavor chemistry that are strong indicators of vineyard-site specific differences in Pinot noir fermentations. 
Such findings provide a concrete starting point for future investigation into site-specific factors that are responsible for wine fermentation outcomes and wine sensory characteristics. 

```{r fig5, echo = F, message = F, warning = F, fig.cap = "Proteomaps depicting the names and pathways of *S. cerevisiae* genes from random forests models that were important for predicting vineyard site. Only the three sites for which *S. cerevisiae* gene expression drove predictive accuracy are depicted. The size of each gene corresponds to the relative importance of that gene for predictive accuracy."}
knitr::include_graphics("figure/sitefig5.pdf")
```

## Conclusion

Microbial biogeography of wine has been documented in globally distributed appellations [@bokulich2014microbial; @pinto2015wine; @wang2015fungal; @garofalo2016non; @bokulich2016associations; @mezzasalma2017grape; @mezzasalma2018geographical; @singh2018genotype; @vitulo2019bark; @liu2020fungal; @abdo2020establishment], and has been correlated with wine fermentation outcomes [@bokulich2016associations; @knight2015regional]. 
In inoculated co-cultures, non-*Saccharomyces* microorganisms both contribute to fermentation and change the behavior of the dominant fermenter *S. cerevisiae*, leading to measurable differences in wine aroma and composition [@tronchoni2017early; @curiel2017different; @alonso2019dominance]. 
Here, we show that grape must DNA profiles do not correlate with the wine transcriptome during primary fermentation. Given that we detected little to no correlation between fungal profiles in initial grape must and genes expressed by those organisms during primary fermentation, DNA profiles may not be an appropriate indicator for wine sensory outcomes in inoculated fermentations. 
However, DNA profiles, in particular bacterial profiles, are predictive of vineyard site and retain signatures of site-specific processes such as total precipitation during the growing season. 
These profiles are rich indicators of the patterns that shape the microbial ecology of grapes, and reflect differences among vineyard sites and regions.

In contrast, the transcriptome profiles of microorganisms, in particular *S. cerevisiae*, retain signatures of vineyard site and region as well as the metabolic transformations that occur during fermentation.
Using *S. cerevisiae* gene expression as a biosensor for differences between fermentations, we detected site specific signatures linked to nitrogen, sulfur, and metal metabolism, with further indicators of potential inter-species interactions with non-*Saccharomyces* organisms.
While these factors are associated with vineyard-specific differences in gene expression profiles, few vineyard site and initial grape must chemical parameters correlate with the transcriptome which suggests there are still many variables to discover that underlie the complex metabolic activities and gene expression patterns *S. cerevisiae* displays throughout fermentation. 
In the future, more direct (a few select sites with characterized differences, e.g. OR1, RRV1, and AS2) and comprehensive sequencing approaches (e.g. deeper sequencing with methods that capture the full transcriptome) aimed at the factors and organisms identified in this work will undoubtedly allow for a better understanding of the processes underlying vineyard-site specific differences in wine fermentations and outcomes. 
This will need to be accompanied by measurements of many more vineyard, must, and wine characteristics to provide further predictive power and insights into the complexities and subtleties of vineyard-site specific wine fermentation outcomes. 

## Methods

### Grape preparation and fermentation

The winemaking protocol has been described previously [@cantu2020investigating; @grainger2020vineyard], but the relevant parts are reproduced with some added details below. 
The grapes used in this study originated from 15 vineyards in eight American Viticulture Areas in California and Oregon.
All grapes were Pinot noir clone 667, while was either rootstock 101-14 (AV1, RRV1, SNC1, SNC2, CRN1, AS1, AS2, SMV1, SMV2, SRH1), Riparia Gloire (OR1, OR2), or 3309C (AV2, RRV2, RRV3). 
We harvested grapes at approximately 25 brix and transported the fruit to University of California, Davis Pilot Winery for fermentation. 
We performed separate fermentations for grapes from each site, with four fermentations for each site (with the exception of AV2 in vintage 2019 where we only performed three fermentations) per vintage. 
After harvest, we separated the fruit into half ton microbins on harvest day and added Inodose SO~2~ to 40 ppm. 
We then moved the bins into a 14C cold room until we destemmed the fruit and divided it into temperature jacket controlled tanks. 
We performed N2 sparging of the tank headspace prior to fermentation, and tanks were sealed with a rubber gasket. 
We performed a cold soak at 7C for three days and adjusted TSO~2~ to 40 ppm on the third day. 
After three days, we increased the fermentation temperature to 21C and programmed pumpover frequency to hold the tank at a constant temperature. 
We took grape must microbiome samples at this time point. 
Then we rehydrated *S. cerevisiae* RC212 with Superstart Rouge at 20 g/hL and inoculated fermentations with 25 g/hL of yeast. 
Approximately 24 hours after inoculation, we adjusted nitrogen content in fermentations using DAP using the formula `(target YAN - 35 mg/L - initial YAN)/2`, and Nutristart using 25 g/hL. 
We only adjusted nitrogen if target YAN was below 250 mg/L in the initial grape must. 
Approximately 48 hours after fermentation, we increased the fermentation temperature to 80F and again added DAP using the formula `(target YAN - 35 mg/L - initial YAN)/2`. 
We ran fermentations to completion, when brix < 0. 
We sampled fermentations for brix at 16 hour and then every twelve hours relative to inoculation and sampled fermentations for RNA at 2 hours, 6 hours (2019 vintage), 16 hours, 64 hours, and 112 hours (2017 and 2019 vintage) relative to inoculation. 
For RNA samples, we performed a ten minute pumpover prior to sampling each tank and sampled 12 mL of wine and centrifuged at 4000 rpm for 5 minutes. 
We discarded the supernatant and froze the pellet in liquid nitrogen, and then stored samples at -80C until RNA extraction.

## Amplicon sequencing data processing

We extracted DNA for amplicon sequencing and performed library preparation following Steenwerth et al. (2020, *in review*). 
The UC Davis DNA Tech Core performed sequencing using Illumina MiSeq, producing 251 base pair paired-end sequences.
We demultiplexed and adapter trimmed libraries by barcode sequences using cutadapt [@martin2011cutadapt].
We produced taxonomically annotated amplicon sequence variant (ASV) counts using DADA2 with the Silva NR database (version 138) for 16S sequences and the UNITE general fasta release (version 8.2) for ITS sequences [@callahan2016dada2].
We filtered all ASVs annotated as "Bacteria,Cyanobacteria,Cyanobacteriia,Chloroplast" and "Bacteria,Proteobacteria,Alphaproteobacteria,Rickettsiales,Mitochondria" as these represent plant mitochondria and chloroplast 16S sequences and not bacterial sequences.

## RNA sequencing data processing

We extracted RNA using the Quick RNA Fungal/Bacterial Miniprep kit including DNAsel column treatment (cat#R2014, Zymo Research). 
We eluted samples in 30 uL of molecular grade water and used 20 uL for sequencing.
We used 3' Tag Sequencing single-end sequencing (Lexogen QuantSeq). 
The UC Davis DNA Technologies Core performed library preparation and sequencing. 
We hard-trimmed the first 12 base pairs from each read and removed Illumina TruSeq adapters and poly(A) tails.
We used sourmash `gather` to determine the organisms present in each sample using parameters `-k 31` and `--scaled 2000` [@brown2016; @pierce2019].
We used the GenBank microbial database (https://sourmash-databases.s3-us-west-2.amazonaws.com/zip/genbank-k31.sbt.zip) as well as a eukaryotic RNA database (https://osf.io/qk5th/). 

Using results from sourmash, we constructed a set of reference genomes representative of all organisms detected within the samples. 
When different strains of the same species are detected, we selected the one species detected in the largest number of samples as a representative species to reduce multi-mapping conflicts. 
We included species present in more than two samples because species present in fewer than three samples would have limited predictive power. 
We also removed species of genus Saccharomyces other than *S. cerevisiae* S288C to reduce multi-mapping conflicts. 
We downloaded selected genomes from NCBI GenBank.
However, if no GTF annotation file was available for the species, we downloaded the genome and GFF3 file from JGI Mycocosm [@grigoriev2014mycocosm], and the GFF3 was converted to GTF using the R package `rtracklayer` [@lawrence2009rtracklayer].
When no annotation file was available on GenBank or JGI Mycocosm, we used the genome of the closest species-level strain with a GTF annotation file.
To find closely related organisms, we searched NCBI taxonomy, selected and downloaded assemblies, and used sourmash `compare` with a k-size of 31 [@brown2016; @pierce2019].
The organisms with the highest Jaccard similarity were considered the most similar.
When no annotation file was available for similar organisms, we generated an annotation file using WebAugustus [@hoff2013webaugustus]. 
See **Figure \@ref(fig:tabs3)** for a description of the best matched genome, the genome used for count generation, and the source of genome annotations.
We concatenated reference genome fasta files and gtf files together to generate a single reference.

```{r tabs3, echo = F,warning=F, message = F, fig.cap = "Species names and accession numbers used to map RNA sequencing reads."}
knitr::include_graphics("figure/sitetabS3.pdf")
```

We then used STAR to align our reads against our constructed reference with parameters `--outFilterType BySJout`, `--outFilterMultimapNmax 20`, `--alignSJoverhangMin 8`, `--alignSJDBoverhangMin 1`, `--outFilterMismatchNmax 999`, `--outFilterMismatchNoverLmax 0.6`, `--alignIntronMin 20`, `--alignIntronMax 1000000`, `--alignMatesGapMax 1000000`, `--outSAMattributes NH HI NM MD` `--outSAMtype BAM`, `SortedByCoordinate` [@dobin2013star].
For the 2019 vintage, we used UMI tools to deduplicate alignments [@smith2017umi].
We quantified reads mapping to each gene using htseq count [@anders2015htseq].

### RC212 genome assembly and comparison

We assembled the *S. cerevisiae* RC212 genome to estimate the fraction of RNA-sequencing reads in each fermentation originating from non-RC212 *S. cerevisiae* strains.
We downloaded fastq files for accession SRR2967888 from the European Nucleotide Archive [@borneman2016whole]. 
We k-mer trimmed reads using the khmer `trim-low-abund.py` command with parameter `-k 20` [@crusoe2015khmer].
We then used the Megahit assembler with default parameters to assemble the reads [@li2015megahit].

### Estimation of non-inoculated yeast in RNA-seq samples

We used sourmash `gather` to estimate the fraction of RNA seq reads (k-mers) originating from non-inoculated *S. cerevisiae*. 
Sourmash `gather` estimates shared sequence similarity by comparing scaled MinHash signatures derived from k-mer profiles [@brown2016; @pierce2019].
We used the sourmash Eukaryotic RNA database (https://osf.io/qk5th/), which includes all annotated *S. cerevisiae* genomes in GenBank (e.g. genomes that include `*rna_from_genome.fna` annotations), as well as our *S. cerevisiae* RC212 genomes assembly.

### ANOSIM and NMDS

We used compositional data analysis for amplicon and transcriptome counts [@gloor2017microbiome].
We used the `transform()` function in the microbiome bioconductor package to transform counts by centered log ratio [@lahti2017tools; @gentleman2004bioconductor]. 
To test for differences among groups, we used Aitchinson distance (Euclidean distance on CLR-transformed counts) and performed the test with the `anosim()` function in the vegan package using parameters `distance = "euclidean"` and `permutations = 9999` [@aitchison2000logratio; @oksanen2007vegan]. 
We used p = 0.05 to indicate statistical significance. 
To construct NMDS plots, we used Aitchinson distance using the `metaMDS()` function in the vegan package with parameter  `distance = "euclidean"`. 
We plotted the results using the ggplot2 package [@wickham2019welcome]. 

### Amplicon sequencing random forest models

We built random forest classifiers using the R ranger package [@wright2015]. 
Using ASV counts produced by DADA2, we first normalized counts by dividing by total number of aligned reads. 
We then used the `tuneRanger()` function in the tuneRanger package to optimize each model for parameters `m.try`, `sample.fraction`, and `min.node.size` [@probst2018tuneranger]. 
We used the `ranger()` function to build each model with parameters from tuneRanger as well as `num.trees = 10000`, `importance = "permutation"`, and `local.importance = TRUE`.
As a supervised technique, random forest classifiers are trained on a subset of data and tested on a separate subset to calculate predictive accuracy.
For models built with samples from all vintages, we used the `createDataPartition()` function in the R caret package to randomly but equally partition training and testing sets with a 70:30 split, ensuring that all class labels were equally represented in both sets [@kuhn2008building]. 
For other models, we built the classifier using all samples from two vintages and validated on the held out vintage. 
We calculated accuracy and kappa statistics for each model. 

### RNA sequencing random forest model

We imported counts into R and normalized by dividing by total number of aligned reads (e.g. library size). 
Given that random forests expects independent samples and RNA-sequencing was conducted in time series over the course of primary fermentation, we summarized each gene from each time series set into mean count, minimum count, maximum count, total count, and standard deviation of counts. 
We performed variable selection using the vita method [@janitza2018].
We then built models using the same methods as with amplicon sequencing models. 

To estimate vineyard-site specific gene importance, we averaged the local importance for each fermentation from a vineyard sites in the training set.
We visualized these gene sets using proteomaps [@liebermeister2014visual], and visualized the intersection of these gene sets using the R package ComplexUpset (https://github.com/krassowski/complex-upset).  
Lastly, we performed gene set enrichment analysis against the Gene Ontology (`ont = "ALL"`) and Kyoto Encyclopedia of Genes and Genomes (`organism = "sce"`) databases using the R package clusterProfiler [@yu2012clusterprofiler].

### Mantel tests

We performed Mantel tests to assess the similarity between samples across measurements bacterial abundance, fungal abundance, transcriptome abundance, initial grape must chemistry, and vineyard site parameters [@mantel1967detection; @legendre2012numerical]. 
The Mantel test determines the correlation between the same samples in different matrices, testing whether similarities between samples estimated from one measurement type match similarities of the same samples estimated from a different measurement type [@mantel1967detection; @legendre2012numerical].
We performed these tests using complete cases, with microbiome and transcriptome abundances from the 2017 and 2019 vintages.
We estimated vineyard site parameters total precipitation and growing degree days using the PRISM climate models including dates April 1 - September 30 in 2017 and 2019 [@daly2000high].
We calculated distance matrices for each matrix using the `dist()` function in R, with parameters `method  = "euclidean"`, with the exception of geographic distance which we calculated using the `distm()` function in the package geosphere with parameter `distHaversine` [@hijmans2017package].
When we calculated distances for disparate measurement types at the same time, we first scaled and centered values using the function `scale()` with parameters `center = TRUE` and `scale = TRUE`. 
We performed Mantel tests with the `mantel()` function in the vegan package with parameters `method = "spearman"`, `permutations = 9999`, and `na.rm = TRUE` [@oksanen2007vegan; @legendre2012numerical]. 
We performed p value adjustment using the function `p.adjust()` with parameter `method = "fdr"`. We used a false discovery rate of p = 0.1. 

<!-- ### Co Authors -->

<!-- Shelby Byer, Isadora Frias, Esmeralda Leon, Robert Viano, Michael Mcloughlin, Thomas Halligan, and Desmon Hernandez, assisted in vinification. -->
<!-- Rachel Montpetit performed DNA and RNA extraction and DNA library preparation, as well as collected some wine samples.  -->
<!-- Kerri Steenwerth, Ron Runnebaum, and Ben Montpetit supervised the work. -->


```{r figs3, echo = F,warning=F, message = F, fig.cap = "Confusion matrices depicting accuracy of random forests models built with RNA sequencing data to predict **A)** vineyard site and **B)** vineyard region. The models depicted were trained on one vintage and validated on the other."}
knitr::include_graphics("figure/sitefigS3.pdf")
```

```{r figs4, echo = F,warning=F, message = F, fig.cap = "Confusion matrices depicting accuracy of random forests models built with bacterial ribosomal DNA amplicon sequencing data to predict **A)** vineyard site and **B)** vineyard region. The models depicted were trained on two vintages and validated on the third."}
knitr::include_graphics("figure/sitefigS4.pdf")
```

```{r figs5, echo = F,warning=F, message = F, fig.cap = "Confusion matrices depicting accuracy of random forests models built with fungal ribosomal DNA amplicon sequencing data to predict **A)** vineyard site and **B)** vineyard region. The models depicted were trained on two vintages and validated on the third."}
knitr::include_graphics("figure/sitefigS5.pdf")
```

```{r figs6, echo = F,warning=F, message = F, fig.cap = "Proteomaps depicting the names and pathways of S. cerevisiae genes from random forests models that were important for predicting vineyard site. Only the three sites for which non-Saccharomyces gene expression drove predictive accuracy are depicted. The size of each gene corresponds to the relative importance of that gene for predictive accuracy."}
knitr::include_graphics("figure/sitefigS6.pdf")
```

